<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="description" content="test description">

<title>5&nbsp; Plague â€“ Interpretable Gene Set Enrichment Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../notebooks/summary.html" rel="next">
<link href="../../notebooks/pbmcs/table.html" rel="prev">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-626149efe8f5d16e1d391ba177679bf0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script src="../../site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="../../site_libs/react-18.2.0/react.min.js"></script>
<script src="../../site_libs/react-18.2.0/react-dom.min.js"></script>
<script src="../../site_libs/reactwidget-2.0.0/react-tools.js"></script>
<link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/reactable-0.4.4/reactable.css" rel="stylesheet">
<script src="../../site_libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="../../site_libs/tippyjs-3.2.0/themes/google.css" rel="stylesheet">
<link href="../../site_libs/tippyjs-3.2.0/themes/light-border.css" rel="stylesheet">
<link href="../../site_libs/tippyjs-3.2.0/themes/light.css" rel="stylesheet">
<link href="../../site_libs/tippyjs-3.2.0/themes/translucent.css" rel="stylesheet">
<script src="../../site_libs/tippyjs-3.2.0/tippy.all.min.js"></script>
<script src="../../site_libs/tippy-binding-0.1.0/tippy.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../notebooks/plague_example/plague.html">Plague analysis</a></li><li class="breadcrumb-item"><a href="../../notebooks/plague_example/plague.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Plague</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">Interpretable Gene Set Enrichment Analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/gsea_simulated_example/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">GSEA with logistic SuSiE, simulated example</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/gseasusie.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">gseasusie</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">PBMC example</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/pbmcs/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">PBMC GSEA Example</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/pbmcs/table.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">PBMC GSEA Table</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Plague analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/plague_example/plague.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Plague</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#functions" id="toc-functions" class="nav-link active" data-scroll-target="#functions"><span class="header-section-number">5.0.1</span> Functions</a></li>
  <li><a href="#de-genes-lfsr-0.01" id="toc-de-genes-lfsr-0.01" class="nav-link" data-scroll-target="#de-genes-lfsr-0.01"><span class="header-section-number">5.1</span> DE genes (<code>lfsr &lt; 0.01</code>)</a>
  <ul class="collapse">
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview"><span class="header-section-number">5.1.1</span> Overview</a></li>
  <li><a href="#fitting" id="toc-fitting" class="nav-link" data-scroll-target="#fitting"><span class="header-section-number">5.1.2</span> Fitting</a></li>
  <li><a href="#b-cell" id="toc-b-cell" class="nav-link" data-scroll-target="#b-cell"><span class="header-section-number">5.1.3</span> B Cell</a></li>
  <li><a href="#cd4t-cell" id="toc-cd4t-cell" class="nav-link" data-scroll-target="#cd4t-cell"><span class="header-section-number">5.1.4</span> CD4T Cell</a></li>
  <li><a href="#cd8t-cell" id="toc-cd8t-cell" class="nav-link" data-scroll-target="#cd8t-cell"><span class="header-section-number">5.1.5</span> CD8T Cell</a></li>
  <li><a href="#mono" id="toc-mono" class="nav-link" data-scroll-target="#mono"><span class="header-section-number">5.1.6</span> Mono</a></li>
  <li><a href="#nk" id="toc-nk" class="nav-link" data-scroll-target="#nk"><span class="header-section-number">5.1.7</span> NK</a></li>
  <li><a href="#residual-enrichments" id="toc-residual-enrichments" class="nav-link" data-scroll-target="#residual-enrichments"><span class="header-section-number">5.1.8</span> Residual enrichments</a></li>
  </ul></li>
  <li><a href="#de-genes-lfsr-0.01-log2fc-0.5" id="toc-de-genes-lfsr-0.01-log2fc-0.5" class="nav-link" data-scroll-target="#de-genes-lfsr-0.01-log2fc-0.5"><span class="header-section-number">5.2</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &gt; 0.5</code>)</a>
  <ul class="collapse">
  <li><a href="#overview-1" id="toc-overview-1" class="nav-link" data-scroll-target="#overview-1"><span class="header-section-number">5.2.1</span> Overview</a></li>
  <li><a href="#fitting-1" id="toc-fitting-1" class="nav-link" data-scroll-target="#fitting-1"><span class="header-section-number">5.2.2</span> Fitting</a></li>
  <li><a href="#b-cell-1" id="toc-b-cell-1" class="nav-link" data-scroll-target="#b-cell-1"><span class="header-section-number">5.2.3</span> B Cell</a></li>
  <li><a href="#cd4t-cell-1" id="toc-cd4t-cell-1" class="nav-link" data-scroll-target="#cd4t-cell-1"><span class="header-section-number">5.2.4</span> CD4T Cell</a></li>
  <li><a href="#cd8t-cell-1" id="toc-cd8t-cell-1" class="nav-link" data-scroll-target="#cd8t-cell-1"><span class="header-section-number">5.2.5</span> CD8T Cell</a></li>
  <li><a href="#mono-1" id="toc-mono-1" class="nav-link" data-scroll-target="#mono-1"><span class="header-section-number">5.2.6</span> Mono</a></li>
  <li><a href="#nk-1" id="toc-nk-1" class="nav-link" data-scroll-target="#nk-1"><span class="header-section-number">5.2.7</span> NK</a></li>
  <li><a href="#residual-enrichment" id="toc-residual-enrichment" class="nav-link" data-scroll-target="#residual-enrichment"><span class="header-section-number">5.2.8</span> Residual Enrichment</a></li>
  </ul></li>
  <li><a href="#de-genes-lfsr-0.01-log2fc-0.5-upregulated" id="toc-de-genes-lfsr-0.01-log2fc-0.5-upregulated" class="nav-link" data-scroll-target="#de-genes-lfsr-0.01-log2fc-0.5-upregulated"><span class="header-section-number">5.3</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &gt; 0.5</code>, upregulated)</a>
  <ul class="collapse">
  <li><a href="#overview-2" id="toc-overview-2" class="nav-link" data-scroll-target="#overview-2"><span class="header-section-number">5.3.1</span> Overview</a></li>
  <li><a href="#fitting-2" id="toc-fitting-2" class="nav-link" data-scroll-target="#fitting-2"><span class="header-section-number">5.3.2</span> Fitting</a></li>
  <li><a href="#b-cell-2" id="toc-b-cell-2" class="nav-link" data-scroll-target="#b-cell-2"><span class="header-section-number">5.3.3</span> B Cell</a></li>
  <li><a href="#cd4t-cell-2" id="toc-cd4t-cell-2" class="nav-link" data-scroll-target="#cd4t-cell-2"><span class="header-section-number">5.3.4</span> CD4T Cell</a></li>
  <li><a href="#cd8t-cell-2" id="toc-cd8t-cell-2" class="nav-link" data-scroll-target="#cd8t-cell-2"><span class="header-section-number">5.3.5</span> CD8T Cell</a></li>
  <li><a href="#mono-2" id="toc-mono-2" class="nav-link" data-scroll-target="#mono-2"><span class="header-section-number">5.3.6</span> Mono</a></li>
  <li><a href="#nk-2" id="toc-nk-2" class="nav-link" data-scroll-target="#nk-2"><span class="header-section-number">5.3.7</span> NK</a></li>
  <li><a href="#residual-enrichment-1" id="toc-residual-enrichment-1" class="nav-link" data-scroll-target="#residual-enrichment-1"><span class="header-section-number">5.3.8</span> Residual Enrichment</a></li>
  </ul></li>
  <li><a href="#de-genes-lfsr-0.01-log2fc--0.5-down-regulated" id="toc-de-genes-lfsr-0.01-log2fc--0.5-down-regulated" class="nav-link" data-scroll-target="#de-genes-lfsr-0.01-log2fc--0.5-down-regulated"><span class="header-section-number">5.4</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &lt; -0.5</code>, down regulated)</a>
  <ul class="collapse">
  <li><a href="#overview-3" id="toc-overview-3" class="nav-link" data-scroll-target="#overview-3"><span class="header-section-number">5.4.1</span> Overview</a></li>
  <li><a href="#fitting-3" id="toc-fitting-3" class="nav-link" data-scroll-target="#fitting-3"><span class="header-section-number">5.4.2</span> Fitting</a></li>
  <li><a href="#b-cell-3" id="toc-b-cell-3" class="nav-link" data-scroll-target="#b-cell-3"><span class="header-section-number">5.4.3</span> B Cell</a></li>
  <li><a href="#cd4t-cell-3" id="toc-cd4t-cell-3" class="nav-link" data-scroll-target="#cd4t-cell-3"><span class="header-section-number">5.4.4</span> CD4T Cell</a></li>
  <li><a href="#cd8t-cell-3" id="toc-cd8t-cell-3" class="nav-link" data-scroll-target="#cd8t-cell-3"><span class="header-section-number">5.4.5</span> CD8T Cell</a></li>
  <li><a href="#mono-3" id="toc-mono-3" class="nav-link" data-scroll-target="#mono-3"><span class="header-section-number">5.4.6</span> Mono</a></li>
  <li><a href="#nk-3" id="toc-nk-3" class="nav-link" data-scroll-target="#nk-3"><span class="header-section-number">5.4.7</span> NK</a></li>
  <li><a href="#residual-enrichment-2" id="toc-residual-enrichment-2" class="nav-link" data-scroll-target="#residual-enrichment-2"><span class="header-section-number">5.4.8</span> Residual Enrichment</a></li>
  </ul></li>
  <li><a href="#unique-de-genes-lfsr-0.01" id="toc-unique-de-genes-lfsr-0.01" class="nav-link" data-scroll-target="#unique-de-genes-lfsr-0.01"><span class="header-section-number">5.5</span> Unique DE genes (<code>lfsr &lt; 0.01</code>)</a>
  <ul class="collapse">
  <li><a href="#overview-4" id="toc-overview-4" class="nav-link" data-scroll-target="#overview-4"><span class="header-section-number">5.5.1</span> Overview</a></li>
  <li><a href="#fitting-4" id="toc-fitting-4" class="nav-link" data-scroll-target="#fitting-4"><span class="header-section-number">5.5.2</span> Fitting</a></li>
  <li><a href="#b-cell-4" id="toc-b-cell-4" class="nav-link" data-scroll-target="#b-cell-4"><span class="header-section-number">5.5.3</span> B Cell</a></li>
  <li><a href="#cd4t-cell-4" id="toc-cd4t-cell-4" class="nav-link" data-scroll-target="#cd4t-cell-4"><span class="header-section-number">5.5.4</span> CD4T Cell</a></li>
  <li><a href="#cd8t-cell-4" id="toc-cd8t-cell-4" class="nav-link" data-scroll-target="#cd8t-cell-4"><span class="header-section-number">5.5.5</span> CD8T Cell</a></li>
  <li><a href="#mono-4" id="toc-mono-4" class="nav-link" data-scroll-target="#mono-4"><span class="header-section-number">5.5.6</span> Mono</a></li>
  <li><a href="#nk-4" id="toc-nk-4" class="nav-link" data-scroll-target="#nk-4"><span class="header-section-number">5.5.7</span> NK</a></li>
  <li><a href="#residual-enrichment-3" id="toc-residual-enrichment-3" class="nav-link" data-scroll-target="#residual-enrichment-3"><span class="header-section-number">5.5.8</span> Residual Enrichment</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../notebooks/plague_example/plague.html">Plague analysis</a></li><li class="breadcrumb-item"><a href="../../notebooks/plague_example/plague.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Plague</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Plague</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Single cell RNA-seq of PBMCs were collected at baseline and stimulated with plaugue. The effect of stimulation was estimated in each celltype using pseudobulk expression. The resutling effect sizes across cell types were restimated using MASH. Differentially expressed genes in a given cell type are identified as those with <span class="math inline">\(\text{lfsr} &lt; 0.01\)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmltools)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tippy)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reactable)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gseasusie)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># reticulate::use_virtualenv('gibss')</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># reticulate::use_virtualenv('~/py/gibss/.venv/')</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">new.env</span>()</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">'data/plague.RData'</span>, <span class="at">env=</span>data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>One thing to note is that the gene lists are quite large when using an lfsr threshold of 0.01, and the DE genes in each cell type have substantial overlap.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>make_intersection_matrix <span class="ot">&lt;-</span> <span class="cf">function</span>(list_of_lists){</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Function to calculate the size of the intersection of two lists</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  intersection_size <span class="ot">&lt;-</span> <span class="cf">function</span>(a, b) {</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(<span class="fu">intersect</span>(a, b))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Determine the number of lists</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_of_lists)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create an empty matrix to store the intersection sizes</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  intersection_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, n, n, <span class="at">dimnames =</span> <span class="fu">list</span>(<span class="fu">names</span>(list_of_lists), <span class="fu">names</span>(list_of_lists)))</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the size of the intersection for each pair</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(list_of_lists)) {</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="fu">seq_along</span>(list_of_lists)) {</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Fill in the matrix with intersection sizes, ensuring i &lt;= j to avoid redundant calculations</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (i <span class="sc">&lt;=</span> j) {</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>        intersection_matrix[i, j] <span class="ot">&lt;-</span> <span class="fu">intersection_size</span>(list_of_lists[[i]], list_of_lists[[j]])</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Since the intersection operation is symmetric, mirror the upper triangle to the lower triangle</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  intersection_matrix[<span class="fu">lower.tri</span>(intersection_matrix)] <span class="ot">&lt;-</span> <span class="fu">t</span>(intersection_matrix)[<span class="fu">lower.tri</span>(intersection_matrix)]</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(intersection_matrix)</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="functions" class="level3" data-number="5.0.1">
<h3 data-number="5.0.1" class="anchored" data-anchor-id="functions"><span class="header-section-number">5.0.1</span> Functions</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>make_cs_tbl <span class="ot">&lt;-</span> gseasusie<span class="sc">:::</span>make_cs_tbl</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>make_component_tbl <span class="ot">&lt;-</span> gseasusie<span class="sc">:::</span>make_component_tbl</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>get_go_info <span class="ot">&lt;-</span> <span class="cf">function</span>(fit){</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(GO.db<span class="sc">::</span>GO.db,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">keys =</span> <span class="fu">unique</span>(fit<span class="sc">$</span>data<span class="sc">$</span>geneSets<span class="sc">$</span>geneSet),</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"TERM"</span>, <span class="st">"DEFINITION"</span>),</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">keytype =</span> <span class="st">"GOID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">geneSet =</span> GOID)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>get_gene_info <span class="ot">&lt;-</span><span class="cf">function</span>(fit){</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db<span class="sc">::</span>org.Hs.eg.db,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">keys =</span> <span class="fu">unique</span>(fit<span class="sc">$</span>data<span class="sc">$</span>geneMapping<span class="sc">$</span>gene),</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"SYMBOL"</span>, <span class="st">"GENETYPE"</span>, <span class="st">"GENENAME"</span>),</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">"ENTREZID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene =</span> ENTREZID)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>make_cs_tbl_nested <span class="ot">&lt;-</span> <span class="cf">function</span>(fit, <span class="at">logbf_threshold=</span><span class="fu">log</span>(<span class="fl">10.</span>)) {</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  more_go_info <span class="ot">&lt;-</span> <span class="fu">get_go_info</span>(fit)</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  more_gene_info <span class="ot">&lt;-</span> <span class="fu">get_gene_info</span>(fit)</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  gene_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>, <span class="st">"GENENAME"</span>, <span class="st">"geneInList"</span>)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  gene_set_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"geneSet"</span>, <span class="st">"TERM"</span>, <span class="st">"DEFINITION"</span>, <span class="st">"alpha"</span>, <span class="st">"beta"</span>, <span class="st">"lbf"</span>, <span class="st">"geneSetSize"</span>, <span class="st">"propInList"</span>)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  component_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"component"</span>, <span class="st">"size"</span>, <span class="st">"top_geneSet"</span>, <span class="st">"top_term"</span>, <span class="st">"top_effect"</span>, <span class="st">"lbf_ser"</span>)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  all_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(component_columns, gene_set_columns, gene_columns)</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">"building nested credible set table"</span>)</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  cs_tbl <span class="ot">&lt;-</span> <span class="fu">make_cs_tbl</span>(fit, logbf_threshold)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>  component_tbl <span class="ot">&lt;-</span> <span class="fu">make_component_tbl</span>(fit, logbf_threshold)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  nested_tbl <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(cs_tbl) <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>    nested_tbl <span class="ot">&lt;-</span> cs_tbl <span class="sc">%&gt;%</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(component_tbl) <span class="sc">%&gt;%</span> <span class="co"># add effect estimates, etc.</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(more_go_info) <span class="sc">%&gt;%</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>      <span class="fu">left_join</span>(more_gene_info) <span class="sc">%&gt;%</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(tidyselect<span class="sc">::</span><span class="fu">any_of</span>(all_columns)) <span class="sc">%&gt;%</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>      <span class="co"># get representative gene set info</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">group_by</span>(component) <span class="sc">%&gt;%</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_geneSet =</span> geneSet[<span class="fu">which.max</span>(lbf)],</span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_term =</span> TERM[<span class="fu">which.max</span>(lbf)],</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>        <span class="at">top_effect =</span> beta[<span class="fu">which.max</span>(lbf)]</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>      <span class="co"># nest gene level data</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">.by =</span> <span class="fu">c</span>(component_columns, gene_set_columns), <span class="at">.key =</span> <span class="st">"details"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>      <span class="co"># nest gene set level data</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">.by =</span> component_columns, <span class="at">.key =</span> <span class="st">"details"</span>)</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(nested_tbl)</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reactable)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># See the ?tippy documentation to learn how to customize tooltips</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>with_tooltip <span class="ot">&lt;-</span> <span class="cf">function</span>(value, tooltip, ...) {</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">div</span>(<span class="at">style =</span> <span class="st">"text-decoration: underline; text-decoration-style: dotted; cursor: help"</span>,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">tippy</span>(value, tooltip, ...))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>make_sensible_colDef <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize an empty list to store colDef objects</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  col_defs <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Iterate over each column in the data</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col_name <span class="cf">in</span> <span class="fu">names</span>(data)) {</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get the column data</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    col_data <span class="ot">&lt;-</span> data[[col_name]]</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Determine the type of the column</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    col_type <span class="ot">&lt;-</span> <span class="fu">class</span>(col_data)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set default colDef based on the column type</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">"numeric"</span> <span class="sc">%in%</span> col_type) {</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Numeric columns: Format to 2 decimal places</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>      col_defs[[col_name]] <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">format =</span> <span class="fu">colFormat</span>(<span class="at">digits =</span> <span class="dv">2</span>))</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"integer"</span> <span class="sc">%in%</span> col_type) {</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Integer columns: No decimal places</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>      col_defs[[col_name]] <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">format =</span> <span class="fu">colFormat</span>(<span class="at">digits =</span> <span class="dv">0</span>))</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"character"</span> <span class="sc">%in%</span> col_type <span class="sc">||</span> <span class="st">"factor"</span> <span class="sc">%in%</span> col_type) {</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Character or factor columns: Truncate long strings</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>      <span class="co"># col_defs[[col_name]] &lt;- colDef(truncate = TRUE, minWidth = 150)</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"Date"</span> <span class="sc">%in%</span> col_type) {</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Date columns: Format as a readable date</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>      col_defs[[col_name]] <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">format =</span> <span class="fu">colFormat</span>(<span class="at">date =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (<span class="st">"logical"</span> <span class="sc">%in%</span> col_type) {</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Logical columns: Display as checkboxes</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>      col_defs[[col_name]] <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">cell =</span> <span class="cf">function</span>(value) {</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">isTRUE</span>(value)) <span class="st">"âœ”"</span> <span class="cf">else</span> <span class="st">"âœ—"</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>      })</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Default colDef for other types</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>      col_defs[[col_name]] <span class="ot">&lt;-</span> <span class="fu">colDef</span>()</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(col_defs)</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>make_rwg_column_style <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">max=</span><span class="dv">2</span>){</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(values){</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>        pal <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">rgb</span>(<span class="fu">colorRamp</span>(<span class="fu">c</span>(<span class="st">'red'</span>, <span class="st">'white'</span>, <span class="st">'green'</span>))(x), <span class="at">maxColorValue=</span><span class="dv">255</span>)</span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>        clipped <span class="ot">&lt;-</span> (<span class="fu">pmax</span>(<span class="sc">-</span>max, <span class="fu">pmin</span>(values, max)) <span class="sc">+</span> max) <span class="sc">/</span> (<span class="dv">2</span><span class="sc">*</span>max)</span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">&lt;-</span> <span class="fu">pal</span>(clipped)</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="at">background =</span> color)</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>make_column_format_list <span class="ot">&lt;-</span> <span class="cf">function</span>(data){</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>  columns <span class="ot">&lt;-</span> <span class="fu">make_sensible_colDef</span>(data)</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">'top_effect'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(data)){</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>    columns<span class="sc">$</span>top_effect <span class="ot">&lt;-</span> reactable<span class="sc">::</span><span class="fu">colDef</span>(</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">style=</span> <span class="fu">make_rwg_column_style</span>(<span class="dv">2</span>),</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">format =</span> reactable<span class="sc">::</span><span class="fu">colFormat</span>(</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits=</span><span class="dv">2</span></span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">'beta'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(data)){</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>    columns<span class="sc">$</span>beta <span class="ot">&lt;-</span> reactable<span class="sc">::</span><span class="fu">colDef</span>(</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>      <span class="at">style=</span> <span class="fu">make_rwg_column_style</span>(<span class="dv">2</span>),</span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>      <span class="at">format =</span> reactable<span class="sc">::</span><span class="fu">colFormat</span>(</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>        <span class="at">digits=</span><span class="dv">2</span></span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">'alpha'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(data)){</span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>    columns<span class="sc">$</span>alpha <span class="ot">&lt;-</span> reactable<span class="sc">::</span><span class="fu">colDef</span>(</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>      <span class="at">format =</span> reactable<span class="sc">::</span><span class="fu">colFormat</span>(<span class="at">digits=</span><span class="dv">2</span>)</span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">'geneSet'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(data)){</span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>    term2def <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(data<span class="sc">$</span>DEFINITION, <span class="sc">~</span><span class="fu">if_else</span>(<span class="fu">is.na</span>(.x), <span class="st">"No definition found"</span>, .x))</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(term2def) <span class="ot">&lt;-</span> data<span class="sc">$</span>geneSet</span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>    columns<span class="sc">$</span>geneSet <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">with_tooltip</span>(value, term2def[[value]]))</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">'top_geneSet'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(data)){</span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>    term2def <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(data<span class="sc">$</span>DEFINITION, <span class="sc">~</span><span class="fu">if_else</span>(<span class="fu">is.na</span>(.x), <span class="st">"No definition found"</span>, .x))</span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(term2def) <span class="ot">&lt;-</span> data<span class="sc">$</span>GOID</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>    columns<span class="sc">$</span>top_geneSet <span class="ot">&lt;-</span> <span class="fu">colDef</span>(<span class="at">cell =</span> <span class="cf">function</span>(value) <span class="fu">with_tooltip</span>(value, term2def[[value]]))</span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(columns)</span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>make_get_details <span class="ot">&lt;-</span> <span class="cf">function</span> (nested_tbls) </span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(nested_tbls)) {</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>      get_details <span class="ot">&lt;-</span> <span class="cf">function</span>(index) {</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>        nested_tbl <span class="ot">&lt;-</span> nested_tbls[[index]]</span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>        data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(nested_tbl, <span class="sc">-</span>tidyselect<span class="sc">::</span><span class="fu">any_of</span>(<span class="st">"details"</span>)) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="st">'details'</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(nested_tbl)){</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>          details <span class="ot">&lt;-</span> nested_tbl<span class="sc">$</span>details</span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span>{</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a>          details <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>        htmltools<span class="sc">::</span><span class="fu">div</span>(<span class="at">style =</span> <span class="st">"padding: 1rem"</span>, reactable<span class="sc">::</span><span class="fu">reactable</span>(</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>          data,</span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>          <span class="at">details =</span> <span class="fu">make_get_details</span>(details), </span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a>          <span class="at">outlined =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a>          <span class="at">columns =</span> <span class="fu">make_column_format_list</span>(data)</span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a>        ))</span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>        get_details <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(get_details)</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>nested_reactable <span class="ot">&lt;-</span> <span class="cf">function</span> (nested_tbl) </span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(nested_tbl)){</span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(nested_tbl, <span class="sc">-</span>tidyselect<span class="sc">::</span><span class="fu">any_of</span>(<span class="st">"details"</span>))</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>  details <span class="ot">&lt;-</span> nested_tbl<span class="sc">$</span>details</span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>  reactable<span class="sc">::</span><span class="fu">reactable</span>(</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>    data, </span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>    <span class="at">details =</span> <span class="fu">make_get_details</span>(details),</span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>    <span class="at">columns =</span> <span class="fu">make_column_format_list</span>(data)</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="de-genes-lfsr-0.01" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="de-genes-lfsr-0.01"><span class="header-section-number">5.1</span> DE genes (<code>lfsr &lt; 0.01</code>)</h2>
<section id="overview" class="level3" data-number="5.1.1">
<h3 data-number="5.1.1" class="anchored" data-anchor-id="overview"><span class="header-section-number">5.1.1</span> Overview</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>make_list_and_background <span class="ot">&lt;-</span> <span class="cf">function</span>(gene, lfsr){</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> gene[<span class="fu">which</span>(lfsr <span class="sc">&lt;</span> <span class="fl">0.01</span>)]</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">gene_list=</span>gene_list, <span class="at">gene_background=</span>gene))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>symbol2entrez <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db, </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">unique</span>(data<span class="sc">$</span>gene_results<span class="sc">$</span>gene), </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">'ENTREZID'</span>, </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">'SYMBOL'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> SYMBOL)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> data<span class="sc">$</span>long_results <span class="sc">%&gt;%</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(symbol2entrez) <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">%&gt;%</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gene_list =</span> <span class="fu">list</span>(<span class="fu">make_list_and_background</span>(ENTREZID, lfsr)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the matrix</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>list_of_lists <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(gene_lists<span class="sc">$</span>gene_list, <span class="sc">~</span>.<span class="sc">$</span>gene_list)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_of_lists) <span class="ot">&lt;-</span> gene_lists<span class="sc">$</span>celltype</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>intersection_matrix <span class="ot">&lt;-</span> <span class="fu">make_intersection_matrix</span>(list_of_lists)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(intersection_matrix, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">display_numbers =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="plague_files/figure-html/heatmap-intersection-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fitting" class="level3" data-number="5.1.2">
<h3 data-number="5.1.2" class="anchored" data-anchor-id="fitting"><span class="header-section-number">5.1.2</span> Fitting</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>susie_fits <span class="ot">&lt;-</span> gene_lists <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fit =</span> <span class="fu">list</span>(gseasusie<span class="sc">::</span><span class="fu">fit_gsea_susie_webgestalt</span>(</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>gene_list, gene_list<span class="sc">$</span>gene_background,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'geneontology_Biological_Process'</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  )))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>8.516 sec elapsed
8.157 sec elapsed
11.921 sec elapsed
17.933 sec elapsed
13.189 sec elapsed</code></pre>
</div>
</div>
</section>
<section id="b-cell" class="level3" data-number="5.1.3">
<h3 data-number="5.1.3" class="anchored" data-anchor-id="b-cell"><span class="header-section-number">5.1.3</span> B Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-29f31364a3deaf480a92" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-29f31364a3deaf480a92">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L3"],"log2BF_SER":[77.7194378448951,29.8665595374129],"geneSet":["GO:0006955","GO:0034097"],"name":["immune response","response to cytokine"],"size":[891,524],"propInList":[0.374859708193042,0.41030534351145],"log2OR":[1.21654765801547,1.08359664174863],"PIP":[0.989356815814972,0.941756069660187],"log2BF":[91.3206632784479,43.3966492092866]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e9165e5031ced195d9f180228fbc00f9"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-4040551da428f329b033","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-4040551da428f329b033"},"children":["{\"x\":{\"opts\":{\"content\":\"Any immune system process that functions in the calibrated response of an organism to a potential internal or invasive threat.\"},\"text\":\"GO:0006955\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-5de91926428e59c1a0fb","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-5de91926428e59c1a0fb"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#63FF63"},{"background":"#74FF74"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"c98536e8fde09d7c2a9668259895a9bb"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>cs_tbl <span class="ot">&lt;-</span> susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>(<span class="at">coverage =</span> <span class="fl">0.99</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>cs_tbl <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(component) <span class="sc">%&gt;%</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(log2BF_SER, DEFINITION)) <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  gt<span class="sc">::</span><span class="fu">gt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  gt<span class="sc">::</span><span class="fu">fmt_number</span>(<span class="at">columns =</span> <span class="fu">c</span>(propInList, log2OR, PIP, log2BF), <span class="at">decimals=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="aaveitjqtj" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#aaveitjqtj table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#aaveitjqtj thead, #aaveitjqtj tbody, #aaveitjqtj tfoot, #aaveitjqtj tr, #aaveitjqtj td, #aaveitjqtj th {
  border-style: none;
}

#aaveitjqtj p {
  margin: 0;
  padding: 0;
}

#aaveitjqtj .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#aaveitjqtj .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#aaveitjqtj .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#aaveitjqtj .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#aaveitjqtj .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aaveitjqtj .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aaveitjqtj .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#aaveitjqtj .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#aaveitjqtj .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#aaveitjqtj .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#aaveitjqtj .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#aaveitjqtj .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#aaveitjqtj .gt_spanner_row {
  border-bottom-style: hidden;
}

#aaveitjqtj .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#aaveitjqtj .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#aaveitjqtj .gt_from_md > :first-child {
  margin-top: 0;
}

#aaveitjqtj .gt_from_md > :last-child {
  margin-bottom: 0;
}

#aaveitjqtj .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#aaveitjqtj .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#aaveitjqtj .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#aaveitjqtj .gt_row_group_first td {
  border-top-width: 2px;
}

#aaveitjqtj .gt_row_group_first th {
  border-top-width: 2px;
}

#aaveitjqtj .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aaveitjqtj .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#aaveitjqtj .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#aaveitjqtj .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aaveitjqtj .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#aaveitjqtj .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#aaveitjqtj .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#aaveitjqtj .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#aaveitjqtj .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#aaveitjqtj .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aaveitjqtj .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aaveitjqtj .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#aaveitjqtj .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#aaveitjqtj .gt_left {
  text-align: left;
}

#aaveitjqtj .gt_center {
  text-align: center;
}

#aaveitjqtj .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#aaveitjqtj .gt_font_normal {
  font-weight: normal;
}

#aaveitjqtj .gt_font_bold {
  font-weight: bold;
}

#aaveitjqtj .gt_font_italic {
  font-style: italic;
}

#aaveitjqtj .gt_super {
  font-size: 65%;
}

#aaveitjqtj .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#aaveitjqtj .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#aaveitjqtj .gt_indent_1 {
  text-indent: 5px;
}

#aaveitjqtj .gt_indent_2 {
  text-indent: 10px;
}

#aaveitjqtj .gt_indent_3 {
  text-indent: 15px;
}

#aaveitjqtj .gt_indent_4 {
  text-indent: 20px;
}

#aaveitjqtj .gt_indent_5 {
  text-indent: 25px;
}

#aaveitjqtj .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#aaveitjqtj div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="geneSet" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">geneSet</th>
<th id="name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">name</th>
<th id="size" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">size</th>
<th id="propInList" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">propInList</th>
<th id="log2OR" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">log2OR</th>
<th id="PIP" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">PIP</th>
<th id="log2BF" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">log2BF</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row odd">
<td colspan="7" id="L1" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">L1</td>
</tr>
<tr class="gt_row_group_first even">
<td class="gt_row gt_left" headers="L1  geneSet">GO:0006955</td>
<td class="gt_row gt_left" headers="L1  name">immune response</td>
<td class="gt_row gt_right" headers="L1  size">891</td>
<td class="gt_row gt_right" headers="L1  propInList">0.37</td>
<td class="gt_row gt_right" headers="L1  log2OR">1.22</td>
<td class="gt_row gt_right" headers="L1  PIP">0.99</td>
<td class="gt_row gt_right" headers="L1  log2BF">91.32</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="L1  geneSet">GO:0002376</td>
<td class="gt_row gt_left" headers="L1  name">immune system process</td>
<td class="gt_row gt_right" headers="L1  size">1292</td>
<td class="gt_row gt_right" headers="L1  propInList">0.33</td>
<td class="gt_row gt_right" headers="L1  log2OR">1.00</td>
<td class="gt_row gt_right" headers="L1  PIP">0.01</td>
<td class="gt_row gt_right" headers="L1  log2BF">84.78</td>
</tr>
<tr class="gt_group_heading_row even">
<td colspan="7" id="L3" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">L3</td>
</tr>
<tr class="gt_row_group_first odd">
<td class="gt_row gt_left" headers="L3  geneSet">GO:0034097</td>
<td class="gt_row gt_left" headers="L3  name">response to cytokine</td>
<td class="gt_row gt_right" headers="L3  size">524</td>
<td class="gt_row gt_right" headers="L3  propInList">0.41</td>
<td class="gt_row gt_right" headers="L3  log2OR">1.08</td>
<td class="gt_row gt_right" headers="L3  PIP">0.94</td>
<td class="gt_row gt_right" headers="L3  log2BF">43.40</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="L3  geneSet">GO:0071345</td>
<td class="gt_row gt_left" headers="L3  name">cellular response to cytokine stimulus</td>
<td class="gt_row gt_right" headers="L3  size">483</td>
<td class="gt_row gt_right" headers="L3  propInList">0.41</td>
<td class="gt_row gt_right" headers="L3  log2OR">1.07</td>
<td class="gt_row gt_right" headers="L3  PIP">0.05</td>
<td class="gt_row gt_right" headers="L3  log2BF">39.10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="L3  geneSet">GO:0070887</td>
<td class="gt_row gt_left" headers="L3  name">cellular response to chemical stimulus</td>
<td class="gt_row gt_right" headers="L3  size">1309</td>
<td class="gt_row gt_right" headers="L3  propInList">0.30</td>
<td class="gt_row gt_right" headers="L3  log2OR">0.68</td>
<td class="gt_row gt_right" headers="L3  PIP">0.00</td>
<td class="gt_row gt_right" headers="L3  log2BF">35.41</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="cd4t-cell" class="level3" data-number="5.1.4">
<h3 data-number="5.1.4" class="anchored" data-anchor-id="cd4t-cell"><span class="header-section-number">5.1.4</span> CD4T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-ebf34cc2c11d7f8e3ecc" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ebf34cc2c11d7f8e3ecc">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2"],"log2BF_SER":[73.8238158701701,12.1492678573277,12.1492678573277],"geneSet":["GO:0002376","GO:0000184","GO:0006614"],"name":["immune system process","nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","SRP-dependent cotranslational protein targeting to membrane"],"size":[1292,114,91],"propInList":[0.647058823529412,0.789473684210526,0.802197802197802],"log2OR":[0.923615628193628,1.80639397593988,1.84579743702701],"PIP":[0.982020914554596,0.884822845458984,0.0345251262187958],"log2BF":[87.4143071329208,25.5893921804394,20.909722236683]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-02efb585a971d49cb7c9","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-02efb585a971d49cb7c9"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"4ada8c9fe906b6ea3f31181dcbdc72ca"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-196f8397d21fdebcfe5d","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-196f8397d21fdebcfe5d"},"children":["{\"x\":{\"opts\":{\"content\":\"The nonsense-mediated decay pathway for nuclear-transcribed mRNAs degrades mRNAs in which an amino-acid codon has changed to a nonsense codon; this prevents the translation of such mRNAs into truncated, and potentially harmful, proteins.\"},\"text\":\"GO:0000184\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"c95fcda773cbad3ca2db6e3f760b3441"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-102c9e5f3624a8b945f6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-102c9e5f3624a8b945f6"},"children":["{\"x\":{\"opts\":{\"content\":\"The targeting of proteins to a membrane that occurs during translation and is dependent upon two key components, the signal-recognition particle (SRP) and the SRP receptor. SRP is a cytosolic particle that transiently binds to the endoplasmic reticulum (ER) signal sequence in a nascent protein, to the large ribosomal unit, and to the SRP receptor in the ER membrane.\"},\"text\":\"GO:0006614\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#89FF89"},{"background":"#18FF18"},{"background":"#13FF13"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"7f513e0524f768249b17f7ac17f13d6e"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd8t-cell" class="level3" data-number="5.1.5">
<h3 data-number="5.1.5" class="anchored" data-anchor-id="cd8t-cell"><span class="header-section-number">5.1.5</span> CD8T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-1dc3606fb7e1d74f3c07" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1dc3606fb7e1d74f3c07">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L4","L4","L4"],"log2BF_SER":[105.143228138374,81.7455335727194,81.7455335727194,29.0335664487862,29.0335664487862,29.0335664487862],"geneSet":["GO:0002376","GO:0006613","GO:0006614","GO:0034097","GO:0071345","GO:0019221"],"name":["immune system process","cotranslational protein targeting to membrane","SRP-dependent cotranslational protein targeting to membrane","response to cytokine","cellular response to cytokine stimulus","cytokine-mediated signaling pathway"],"size":[1292,95,91,524,483,325],"propInList":[0.400928792569659,0.778947368421053,0.791208791208791,0.473282442748092,0.480331262939959,0.529230769230769],"log2OR":[1.12607093905824,3.56738556394789,3.64916881787798,1.03383036603376,1.06871188728175,1.25997112757519],"PIP":[0.996101379394531,0.570400476455688,0.426819264888763,0.491222858428955,0.365465223789215,0.143237844109535],"log2BF":[118.754257068897,94.5522420536618,94.1338921916462,41.6246803126394,41.1980340075461,39.8467154128524]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-1bd185a0c1dcb94722ca","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-1bd185a0c1dcb94722ca"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e6bbf7029f98bc51cc372c06402527c8"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e1b2984ce297a724bc71","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e1b2984ce297a724bc71"},"children":["{\"x\":{\"opts\":{\"content\":\"The targeting of proteins to a membrane that occurs during translation. The transport of most secretory proteins, particularly those with more than 100 amino acids, into the endoplasmic reticulum lumen occurs in this manner, as does the import of some proteins into mitochondria.\"},\"text\":\"GO:0006613\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"c95fcda773cbad3ca2db6e3f760b3441"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-4062ae716ec2bfd241ad","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-4062ae716ec2bfd241ad"},"children":["{\"x\":{\"opts\":{\"content\":\"The targeting of proteins to a membrane that occurs during translation and is dependent upon two key components, the signal-recognition particle (SRP) and the SRP receptor. SRP is a cytosolic particle that transiently binds to the endoplasmic reticulum (ER) signal sequence in a nascent protein, to the large ribosomal unit, and to the SRP receptor in the ER membrane.\"},\"text\":\"GO:0006614\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-f741a657f35e91ca81e6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-f741a657f35e91ca81e6"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"f4c34bdee6ec9d98a0cb7a54d5552e5d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-2dc5cac6b6ed1b93886a","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-2dc5cac6b6ed1b93886a"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0071345\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-6c1892ad98fc5c35adc0","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-6c1892ad98fc5c35adc0"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#6FFF6F"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#7BFF7B"},{"background":"#76FF76"},{"background":"#5EFF5E"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"2148247c6b72d3027ea6a56cf1186c3e"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="mono" class="level3" data-number="5.1.6">
<h3 data-number="5.1.6" class="anchored" data-anchor-id="mono"><span class="header-section-number">5.1.6</span> Mono</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">4</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-5ff59f668f7a35048d93" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5ff59f668f7a35048d93">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L2","L2","L2"],"log2BF_SER":[94.2001369538037,14.5625020882103,14.5625020882103,14.5625020882103,14.5625020882103,14.5625020882103],"geneSet":["GO:0050896","GO:0090304","GO:0043170","GO:2000112","GO:0044260","GO:0031326"],"name":["response to stimulus","nucleic acid metabolic process","macromolecule metabolic process",null,null,"regulation of cellular biosynthetic process"],"size":[3368,2817,4589,1963,4121,2073],"propInList":[0.790676959619953,0.710330138445154,0.724122902593158,0.709628120224147,0.727250667313759,0.712493970091655],"log2OR":[0.72880116254155,-0.411391572948671,-0.316749477183205,-0.453304604877192,-0.319160243991674,-0.440799703337295],"PIP":[0.999992370605469,0.67740797996521,0.157434418797493,0.0247608125209808,0.0223818011581898,0.0213364847004414],"log2BF":[107.816787866733,27.6172630064033,25.5119889806614,22.8433685837534,22.6976359888571,22.6286324217473]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6b08ea515cd96f80677742cd25f2f7fb"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-a0c8c3418c5e89ed27ca","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-a0c8c3418c5e89ed27ca"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus. The process begins with detection of the stimulus and ends with a change in state or activity or the cell or organism.\"},\"text\":\"GO:0050896\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"075937d009e9be46a0b1362616e79804"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-01ebbfab8a1f046edc77","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-01ebbfab8a1f046edc77"},"children":["{\"x\":{\"opts\":{\"content\":\"Any cellular metabolic process involving nucleic acids.\"},\"text\":\"GO:0090304\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6134a0aa36d94f0b87249e69a57cff01"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-574d8b886b6f3103899c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-574d8b886b6f3103899c"},"children":["{\"x\":{\"opts\":{\"content\":\"The chemical reactions and pathways involving macromolecules, any molecule of high relative molecular mass, the structure of which essentially comprises the multiple repetition of units derived, actually or conceptually, from molecules of low relative molecular mass.\"},\"text\":\"GO:0043170\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"bc65068866681d204c43d3d87652bf1d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-1b091591ff1b6c64df59","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-1b091591ff1b6c64df59"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:2000112\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"56fa837d8abffb40c9a749fd2628076b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-2d3b1c4f56a9d9b596bd","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-2d3b1c4f56a9d9b596bd"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:0044260\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"8a0e111c498755eeee7f54f13d1f8414"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-f1b29143e2adcae5ad11","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-f1b29143e2adcae5ad11"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that modulates the frequency, rate or extent of the chemical reactions and pathways resulting in the formation of substances, carried out by individual cells.\"},\"text\":\"GO:0031326\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#A2FFA2"},{"background":"#FFCACA"},{"background":"#FFD6D6"},{"background":"#FFC5C5"},{"background":"#FFD6D6"},{"background":"#FFC6C6"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"6e30a83924184090a2ef11e05a976bc8"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="nk" class="level3" data-number="5.1.7">
<h3 data-number="5.1.7" class="anchored" data-anchor-id="nk"><span class="header-section-number">5.1.7</span> NK</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">5</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-c1f9a2768b5c706823fe" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c1f9a2768b5c706823fe">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L3","L3","L4","L4","L4","L4","L4","L4","L4","L4","L4"],"log2BF_SER":[99.3883061965595,36.4397347375431,36.4397347375431,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671,5.94359153571671],"geneSet":["GO:0002376","GO:0034097","GO:0071345","GO:0034641","GO:0006139","GO:0046483","GO:1901360","GO:0006725","GO:1901576","GO:0009058","GO:0044238","GO:0006807"],"name":["immune system process","response to cytokine","cellular response to cytokine stimulus",null,"nucleobase-containing compound metabolic process",null,null,null,null,"biosynthetic process","primary metabolic process",null],"size":[1292,524,483,3510,3140,3188,3264,3195,3098,3119,5088,4918],"propInList":[0.371517027863777,0.446564885496183,0.451345755693582,0.188888888888889,0.187579617834395,0.188205771643664,0.189031862745098,0.188732394366197,0.196901226597805,0.197819814042962,0.200078616352201,0.200284668564457],"log2OR":[1.12103081549205,1.13468262633489,1.15492739209856,-0.343430715153242,-0.363058944611394,-0.358212989458033,-0.349096533101097,-0.352763138723998,-0.339315772257373,-0.333713265422866,-0.260730019772121,-0.264626695186214],"PIP":[0.979970157146454,0.811833620071411,0.182943165302277,0.203715071082115,0.200185477733612,0.171848207712173,0.11698941141367,0.115272141993046,0.0393823347985744,0.0284296739846468,0.0211927238851786,0.0209076628088951],"log2BF":[112.975783089472,49.7556540663519,47.605865427122,17.2648801583355,17.2396647499966,17.0194622914154,16.4647054217639,16.4433713178897,14.8939479714589,14.4237969350034,13.9999683936619,13.9804311645239]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-96161bd14c3e4f0a53b3","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-96161bd14c3e4f0a53b3"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-65c73312c0e3ed4445ff","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-65c73312c0e3ed4445ff"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"f4c34bdee6ec9d98a0cb7a54d5552e5d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-d57ccdbb3b4b58a14025","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-d57ccdbb3b4b58a14025"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0071345\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"62902407e5d9991c307b64ab7ee4a07d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-48601a4a53ba82d62ed0","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-48601a4a53ba82d62ed0"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:0034641\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"c48c2d58af4fef223ca619c6f80dd288"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-267cde478c9602ea8a00","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-267cde478c9602ea8a00"},"children":["{\"x\":{\"opts\":{\"content\":\"Any cellular metabolic process involving nucleobases, nucleosides, nucleotides and nucleic acids.\"},\"text\":\"GO:0006139\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"c5f89e60e61cd050fd1eb05ba5dcc57d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-67f1e2792924d3f7319f","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-67f1e2792924d3f7319f"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:0046483\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"1dc93721b5c5c2e293180e48abe0a2e9"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e3c23f7e342379870841","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e3c23f7e342379870841"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:1901360\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ae3110c9d6f2a10cdb85fb68bafdf97e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-7b3a557bf1b2946044fb","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-7b3a557bf1b2946044fb"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:0006725\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e4c765a6d3c328b3ae94030b1dd21ab0"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-9aea0ae524f79799689c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-9aea0ae524f79799689c"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:1901576\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"255148e52f749b9cf4d7c0fd218b807b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-d04bb445fb7945f90f45","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-d04bb445fb7945f90f45"},"children":["{\"x\":{\"opts\":{\"content\":\"The chemical reactions and pathways resulting in the formation of substances; typically the energy-requiring part of metabolism in which simpler substances are transformed into more complex ones.\"},\"text\":\"GO:0009058\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"b71a5dc7810424ab1abed4d0a2ca0b24"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-529621b6e47ff5aa2bbc","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-529621b6e47ff5aa2bbc"},"children":["{\"x\":{\"opts\":{\"content\":\"The chemical reactions and pathways involving those compounds which are formed as a part of the normal anabolic and catabolic processes. These processes take place in most, if not all, cells of the organism.\"},\"text\":\"GO:0044238\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"b52c1b6a5ede5c277758b452047b096d"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-872879a3880d0b3bc66e","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-872879a3880d0b3bc66e"},"children":["{\"x\":{\"opts\":{\"content\":\"No definition found\"},\"text\":\"GO:0006807\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#70FF70"},{"background":"#6EFF6E"},{"background":"#6BFF6B"},{"background":"#FFD3D3"},{"background":"#FFD0D0"},{"background":"#FFD1D1"},{"background":"#FFD2D2"},{"background":"#FFD2D2"},{"background":"#FFD3D3"},{"background":"#FFD4D4"},{"background":"#FFDDDD"},{"background":"#FFDDDD"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"f7d4ac6645552dc1982c025262bb180f"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="residual-enrichments" class="level3" data-number="5.1.8">
<h3 data-number="5.1.8" class="anchored" data-anchor-id="residual-enrichments"><span class="header-section-number">5.1.8</span> Residual enrichments</h3>
<p>While logistic SuSiE may not confidently include a gene set in a CS, there may be evidence for residual enrichment signal after controlling for the effects estimated by SuSiE. That is, we may find examples where there is evidence for a <em>an</em> enrichment (the residual SER has a moderately large BF) but it is difficult to localize to a particular gene set (the CS is very large).</p>
<p>One idea is to perform a marginal analysis of the residual enrichment signal. For each gene set, test for an effect controlling for those discovered in logistic SuSiE. We can apply BH to <span class="math inline">\(p_i = 1/\text{BF}_i\)</span> where <span class="math inline">\(\text{BF}_i\)</span> is the Bayes factor for the enrichment of gene set <span class="math inline">\(i\)</span> controlling for the gene sets discovered in logistic SuSiE. Notice that</p>
<p><span class="math display">\[
\text{BF}_{SER} = \frac{1}{p} \sum_{i=1}^p BF_i \leq \frac{\text{BF}_{(1)}}{p}
\]</span> Then, if <span class="math inline">\(\text{BF}_{SER} &gt; \frac{1}{\alpha}\)</span>, applying BH procedure to <span class="math inline">\(p_i = 1/\text{BH_i}\)</span> would give at least one gene set at FDR level <span class="math inline">\(\alpha\)</span>. For <span class="math inline">\(\alpha = 0.1\)</span> this corresponds to seeing a <span class="math inline">\(\log BF \approx 2.3\)</span>. Here, we present all CSs for components with <span class="math inline">\(\text{BF}_{SER} \geq 10\)</span>. All the CSs are small, so in this example there isnâ€™t evidence for strong marginal enrichment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">map_dbl</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(<span class="dv">1</span><span class="sc">/</span><span class="fu">exp</span>(susie_fits<span class="sc">$</span>fit[[.x]]<span class="sc">$</span>fit<span class="sc">$</span>lbf[<span class="dv">10</span>,]), <span class="at">method=</span><span class="st">'BH'</span>)<span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0 0 0 0 0</code></pre>
</div>
</div>
</section>
</section>
<section id="de-genes-lfsr-0.01-log2fc-0.5" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="de-genes-lfsr-0.01-log2fc-0.5"><span class="header-section-number">5.2</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &gt; 0.5</code>)</h2>
<section id="overview-1" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="overview-1"><span class="header-section-number">5.2.1</span> Overview</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>make_list_and_background <span class="ot">&lt;-</span> <span class="cf">function</span>(gene, lfsr, beta){</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> gene[<span class="fu">intersect</span>(<span class="fu">which</span>(lfsr <span class="sc">&lt;</span> <span class="fl">0.01</span>), <span class="fu">which</span>(<span class="fu">abs</span>(beta) <span class="sc">&gt;</span> <span class="fl">0.5</span>))]</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">gene_list=</span>gene_list, <span class="at">gene_background=</span>gene))</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>symbol2entrez <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db, </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">unique</span>(data<span class="sc">$</span>gene_results<span class="sc">$</span>gene), </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">'ENTREZID'</span>, </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">'SYMBOL'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> SYMBOL)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> data<span class="sc">$</span>long_results <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(symbol2entrez) <span class="sc">%&gt;%</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gene_list =</span> <span class="fu">list</span>(<span class="fu">make_list_and_background</span>(ENTREZID, lfsr, beta)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the matrix</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>list_of_lists <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(gene_lists<span class="sc">$</span>gene_list, <span class="sc">~</span>.<span class="sc">$</span>gene_list)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_of_lists) <span class="ot">&lt;-</span> gene_lists<span class="sc">$</span>celltype</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>intersection_matrix <span class="ot">&lt;-</span> <span class="fu">make_intersection_matrix</span>(list_of_lists)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(intersection_matrix, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">display_numbers =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="plague_files/figure-html/heatmap-intersection-large-effect-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fitting-1" class="level3" data-number="5.2.2">
<h3 data-number="5.2.2" class="anchored" data-anchor-id="fitting-1"><span class="header-section-number">5.2.2</span> Fitting</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>susie_fits <span class="ot">&lt;-</span> gene_lists <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fit =</span> <span class="fu">list</span>(gseasusie<span class="sc">::</span><span class="fu">fit_gsea_susie_webgestalt</span>(</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>gene_list, gene_list<span class="sc">$</span>gene_background,</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'geneontology_Biological_Process'</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  )))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>18.015 sec elapsed
11.357 sec elapsed
14.121 sec elapsed
8.752 sec elapsed
10.563 sec elapsed</code></pre>
</div>
</div>
</section>
<section id="b-cell-1" class="level3" data-number="5.2.3">
<h3 data-number="5.2.3" class="anchored" data-anchor-id="b-cell-1"><span class="header-section-number">5.2.3</span> B Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-4fe62a5b9ed9a08f68f7" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4fe62a5b9ed9a08f68f7">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L2"],"log2BF_SER":[77.3554515113182,69.4931326077035,69.4931326077035,69.4931326077035],"geneSet":["GO:0019221","GO:0009607","GO:0051707","GO:0043207"],"name":["cytokine-mediated signaling pathway","response to biotic stimulus","response to other organism","response to external biotic stimulus"],"size":[325,390,375,375],"propInList":[0.227692307692308,0.197435897435897,0.202666666666667,0.202666666666667],"log2OR":[2.70636439739698,2.47071428513272,2.49137112132785,2.49130662783377],"PIP":[0.996895849704742,0.370791614055634,0.313543617725372,0.313414484262466],"log2BF":[90.9676282996464,81.6784793474651,81.4365369059537,81.4359425339123]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-4d9e12615800d035f066","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-4d9e12615800d035f066"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2149a3af55e2d385df4b0510f2e7fd58"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e967252bdac8c11b40be","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e967252bdac8c11b40be"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a biotic stimulus, a stimulus caused or produced by a living organism.\"},\"text\":\"GO:0009607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3d096c095c85e6af7f3a440747b5ea14"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-ab3abd869be07f7f2e4f","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-ab3abd869be07f7f2e4f"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from another living organism.\"},\"text\":\"GO:0051707\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"268cbbdc024d4488d878e89ff1b9523b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-d4715e1df08f8133df44","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-d4715e1df08f8133df44"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of an external biotic stimulus, an external stimulus caused by, or produced by living things.\"},\"text\":\"GO:0043207\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#00FF00"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"e2a8a1e6b3e0654ffe80c52a08419e31"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd4t-cell-1" class="level3" data-number="5.2.4">
<h3 data-number="5.2.4" class="anchored" data-anchor-id="cd4t-cell-1"><span class="header-section-number">5.2.4</span> CD4T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-ed2100529122fca3d85e" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ed2100529122fca3d85e">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L4","L4","L4","L4"],"log2BF_SER":[72.4246240686753,35.930683161246,35.930683161246,21.8298263566324,21.8298263566324,21.8298263566324,21.8298263566324],"geneSet":["GO:0002376","GO:0019221","GO:0034097","GO:0051607","GO:0009615","GO:0051707","GO:0043207"],"name":["immune system process","cytokine-mediated signaling pathway","response to cytokine","defense response to virus","response to virus","response to other organism","response to external biotic stimulus"],"size":[1292,325,524,140,186,375,375],"propInList":[0.127708978328173,0.249230769230769,0.194656488549618,0.371428571428571,0.317204301075269,0.232,0.232],"log2OR":[1.57655980097984,1.8143915131703,1.54524623180665,2.00024009327627,1.82722055901141,1.38229112227028,1.38219575789037],"PIP":[0.999736845493317,0.610564112663269,0.336807578802109,0.424506604671478,0.327474415302277,0.0806960240006447,0.0806914046406746],"log2BF":[86.0409059658563,48.835560642848,47.9773424495408,34.2103486340437,33.8359437789766,31.8151311210041,31.8150485693317]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-21b857702be93fc27a77","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-21b857702be93fc27a77"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-0a2de761a5e32dffe132","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-0a2de761a5e32dffe132"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-cdc56af4733225b63d5f","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-cdc56af4733225b63d5f"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"dece375ce5a8369e1b6f2f6ad6be3d7e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-b0ebcb531a0cd8d21b14","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-b0ebcb531a0cd8d21b14"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions triggered in response to the presence of a virus that act to protect the cell or organism.\"},\"text\":\"GO:0051607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6bd9d741ac345ee04e5e90714066e6a4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e4237476856bede50bb0","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e4237476856bede50bb0"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from a virus.\"},\"text\":\"GO:0009615\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3d096c095c85e6af7f3a440747b5ea14"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-cfe7bea6f2b8be5043e7","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-cfe7bea6f2b8be5043e7"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from another living organism.\"},\"text\":\"GO:0051707\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"268cbbdc024d4488d878e89ff1b9523b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-510501652b679e1058fb","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-510501652b679e1058fb"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of an external biotic stimulus, an external stimulus caused by, or produced by living things.\"},\"text\":\"GO:0043207\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#35FF35"},{"background":"#17FF17"},{"background":"#39FF39"},{"background":"#00FF00"},{"background":"#16FF16"},{"background":"#4EFF4E"},{"background":"#4EFF4E"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"d1e2b8086f9fdab7b25c83d2389fe9f3"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd8t-cell-1" class="level3" data-number="5.2.5">
<h3 data-number="5.2.5" class="anchored" data-anchor-id="cd8t-cell-1"><span class="header-section-number">5.2.5</span> CD8T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-6511fbb9cf19facc859b" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6511fbb9cf19facc859b">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L4","L4"],"log2BF_SER":[68.9026830336893,39.4617554680459,39.4617554680459,21.7758812242392,21.7758812242392],"geneSet":["GO:0006952","GO:0034097","GO:0019221","GO:0051607","GO:0009615"],"name":["defense response","response to cytokine","cytokine-mediated signaling pathway","defense response to virus","response to virus"],"size":[583,524,325,140,186],"propInList":[0.169811320754717,0.158396946564885,0.206153846153846,0.35,0.290322580645161],"log2OR":[2.11248921389346,1.83897522923259,2.05677801388622,2.0686344978367,1.91527344047871],"PIP":[0.97440105676651,0.817344605922699,0.118702605366707,0.620967984199524,0.376345306634903],"log2BF":[82.4819327612927,52.7874357805443,50.0038429177426,34.7051358412699,33.9826738732479]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"52bc1ab491674748aee36c3369832e5b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-8e784ccb459c9eeb4aaf","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-8e784ccb459c9eeb4aaf"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions, triggered in response to the presence of a foreign body or the occurrence of an injury, which result in restriction of damage to the organism attacked or prevention/recovery from the infection caused by the attack.\"},\"text\":\"GO:0006952\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-32c457b5f2d9b47b1f0a","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-32c457b5f2d9b47b1f0a"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-0988ed02442bec6bfd7d","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-0988ed02442bec6bfd7d"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"dece375ce5a8369e1b6f2f6ad6be3d7e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-6b9ce6e2caf2af85cb30","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-6b9ce6e2caf2af85cb30"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions triggered in response to the presence of a virus that act to protect the cell or organism.\"},\"text\":\"GO:0051607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6bd9d741ac345ee04e5e90714066e6a4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-1edaabab58366689205a","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-1edaabab58366689205a"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from a virus.\"},\"text\":\"GO:0009615\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#00FF00"},{"background":"#14FF14"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#0AFF0A"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"629845a0a31447df9aee7124a57ea820"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="mono-1" class="level3" data-number="5.2.6">
<h3 data-number="5.2.6" class="anchored" data-anchor-id="mono-1"><span class="header-section-number">5.2.6</span> Mono</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">4</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-5c2e6fe58987500f9eb4" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5c2e6fe58987500f9eb4">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L3","L3"],"log2BF_SER":[37.4464417777181,18.3562519892435,18.3562519892435,22.5399752457633,22.5399752457633],"geneSet":["GO:0006952","GO:0022613","GO:0042254","GO:0007166","GO:0007165"],"name":["defense response","ribonucleoprotein complex biogenesis","ribosome biogenesis","cell surface receptor signaling pathway","signal transduction"],"size":[583,381,239,1064,2123],"propInList":[0.361921097770154,0.057742782152231,0.0334728033472803,0.296052631578947,0.250588789448893],"log2OR":[1.14003352255119,-1.74927784965774,-2.3457324289589,0.759223259643998,0.536200027614752],"PIP":[0.966230094432831,0.761453032493591,0.220367267727852,0.877299308776855,0.0601160973310471],"log2BF":[51.0135434299297,31.5797432822824,29.7908980720712,35.9677801811788,32.1005342663191]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"52bc1ab491674748aee36c3369832e5b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-61bc01d1d06e35e40d36","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-61bc01d1d06e35e40d36"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions, triggered in response to the presence of a foreign body or the occurrence of an injury, which result in restriction of damage to the organism attacked or prevention/recovery from the infection caused by the attack.\"},\"text\":\"GO:0006952\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"504da2f9c5f762c92179bc3af66f7d04"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-c4e52d2d13d04fa75f86","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-c4e52d2d13d04fa75f86"},"children":["{\"x\":{\"opts\":{\"content\":\"A cellular process that results in the biosynthesis of constituent macromolecules, assembly, and arrangement of constituent parts of a complex containing RNA and proteins. Includes the biosynthesis of the constituent RNA and protein molecules, and those macromolecular modifications that are involved in synthesis or assembly of the ribonucleoprotein complex.\"},\"text\":\"GO:0022613\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"1a5cc7e3ee3e098441fd00ae3b4cf817"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-6d14f2b3b062b951a868","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-6d14f2b3b062b951a868"},"children":["{\"x\":{\"opts\":{\"content\":\"A cellular process that results in the biosynthesis of constituent macromolecules, assembly, and arrangement of constituent parts of ribosome subunits; includes transport to the sites of protein synthesis.\"},\"text\":\"GO:0042254\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"fabb51113a379de69a6adc2745cdc601"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-7b5879659045fcfacea7","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-7b5879659045fcfacea7"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by an extracellular ligand binding to a receptor located on the cell surface. The pathway ends with regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0007166\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3f81b6762a3e43cf473e4c3ac3a48dc8"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-9bbec8b92622a4f49d36","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-9bbec8b92622a4f49d36"},"children":["{\"x\":{\"opts\":{\"content\":\"The cellular process in which a signal is conveyed to trigger a change in the activity or state of a cell. Signal transduction begins with reception of a signal (e.g. a ligand binding to a receptor or receptor activation by a stimulus such as light), or for signal transduction in the absence of ligand, signal-withdrawal or the activity of a constitutively active receptor. Signal transduction ends with regulation of a downstream cellular process, e.g. regulation of transcription or regulation of a metabolic process. Signal transduction covers signaling from receptors located on the surface of the cell and signaling via molecules located within the cell. For signaling between cells, signal transduction is restricted to events at and within the receiving cell.\"},\"text\":\"GO:0007165\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#6DFF6D"},{"background":"#FF1F1F"},{"background":"#FF0000"},{"background":"#9EFF9E"},{"background":"#BAFFBA"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"3e194df0969b76b612ca43f1994ad462"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="nk-1" class="level3" data-number="5.2.7">
<h3 data-number="5.2.7" class="anchored" data-anchor-id="nk-1"><span class="header-section-number">5.2.7</span> NK</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">5</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-09aac6e360f675f4b8a3" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-09aac6e360f675f4b8a3">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L2","L4"],"log2BF_SER":[35.3880512460148,35.3880512460148,45.3982752960202,44.5037841608712],"geneSet":["GO:0006955","GO:0002376","GO:0034097","GO:0051607"],"name":["immune response","immune system process","response to cytokine","defense response to virus"],"size":[891,1292,524,140],"propInList":[0.129068462401796,0.106037151702786,0.173664122137405,0.385714285714286],"log2OR":[1.45873792652675,1.29758433525817,1.87664837286939,2.65601252075593],"PIP":[0.500873148441315,0.443542540073395,0.95622718334198,0.948257505893707],"log2BF":[48.0072316583996,47.831858885523,58.9503639052915,58.0437979491738]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e9165e5031ced195d9f180228fbc00f9"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-2e62217a910f6bd402d6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-2e62217a910f6bd402d6"},"children":["{\"x\":{\"opts\":{\"content\":\"Any immune system process that functions in the calibrated response of an organism to a potential internal or invasive threat.\"},\"text\":\"GO:0006955\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-ae3a9d5396d58faebad6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-ae3a9d5396d58faebad6"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-40816261c34204e81427","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-40816261c34204e81427"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"dece375ce5a8369e1b6f2f6ad6be3d7e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-bfdec4e45f60efa83875","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-bfdec4e45f60efa83875"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions triggered in response to the presence of a virus that act to protect the cell or organism.\"},\"text\":\"GO:0051607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#45FF45"},{"background":"#59FF59"},{"background":"#0FFF0F"},{"background":"#00FF00"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"5be4d327a1faf8e27e2b56fd9041affb"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="residual-enrichment" class="level3" data-number="5.2.8">
<h3 data-number="5.2.8" class="anchored" data-anchor-id="residual-enrichment"><span class="header-section-number">5.2.8</span> Residual Enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">map_dbl</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(<span class="dv">1</span><span class="sc">/</span><span class="fu">exp</span>(susie_fits<span class="sc">$</span>fit[[.x]]<span class="sc">$</span>fit<span class="sc">$</span>lbf[<span class="dv">10</span>,]), <span class="at">method=</span><span class="st">'BH'</span>)<span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0 0 0 0 0</code></pre>
</div>
</div>
</section>
</section>
<section id="de-genes-lfsr-0.01-log2fc-0.5-upregulated" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="de-genes-lfsr-0.01-log2fc-0.5-upregulated"><span class="header-section-number">5.3</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &gt; 0.5</code>, upregulated)</h2>
<section id="overview-2" class="level3" data-number="5.3.1">
<h3 data-number="5.3.1" class="anchored" data-anchor-id="overview-2"><span class="header-section-number">5.3.1</span> Overview</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>make_list_and_background <span class="ot">&lt;-</span> <span class="cf">function</span>(gene, lfsr, beta){</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> gene[<span class="fu">intersect</span>(<span class="fu">which</span>(lfsr <span class="sc">&lt;</span> <span class="fl">0.01</span>), <span class="fu">which</span>(beta <span class="sc">&gt;</span> <span class="fl">0.5</span>))]</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">gene_list=</span>gene_list, <span class="at">gene_background=</span>gene))</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>symbol2entrez <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db, </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">unique</span>(data<span class="sc">$</span>gene_results<span class="sc">$</span>gene), </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">'ENTREZID'</span>, </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">'SYMBOL'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> SYMBOL)</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> data<span class="sc">$</span>long_results <span class="sc">%&gt;%</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(symbol2entrez) <span class="sc">%&gt;%</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">%&gt;%</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gene_list =</span> <span class="fu">list</span>(<span class="fu">make_list_and_background</span>(ENTREZID, lfsr, beta)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the matrix</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>list_of_lists <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(gene_lists<span class="sc">$</span>gene_list, <span class="sc">~</span>.<span class="sc">$</span>gene_list)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_of_lists) <span class="ot">&lt;-</span> gene_lists<span class="sc">$</span>celltype</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>intersection_matrix <span class="ot">&lt;-</span> <span class="fu">make_intersection_matrix</span>(list_of_lists)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(intersection_matrix, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">display_numbers =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="plague_files/figure-html/heatmap-intersection-large-effect-up-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fitting-2" class="level3" data-number="5.3.2">
<h3 data-number="5.3.2" class="anchored" data-anchor-id="fitting-2"><span class="header-section-number">5.3.2</span> Fitting</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>susie_fits <span class="ot">&lt;-</span> gene_lists <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fit =</span> <span class="fu">list</span>(gseasusie<span class="sc">::</span><span class="fu">fit_gsea_susie_webgestalt</span>(</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>gene_list, gene_list<span class="sc">$</span>gene_background,</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'geneontology_Biological_Process'</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  )))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>8.59 sec elapsed
12.306 sec elapsed
9.439 sec elapsed
10.195 sec elapsed
10.55 sec elapsed</code></pre>
</div>
</div>
</section>
<section id="b-cell-2" class="level3" data-number="5.3.3">
<h3 data-number="5.3.3" class="anchored" data-anchor-id="b-cell-2"><span class="header-section-number">5.3.3</span> B Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-5eaa3bbdea091d4e11af" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5eaa3bbdea091d4e11af">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L2","L3","L3"],"log2BF_SER":[111.897796507845,26.4314324815825,26.4314324815825,26.4314324815825,5.56707446122771,5.56707446122771],"geneSet":["GO:0006952","GO:0034340","GO:0071357","GO:0060337","GO:0006396","GO:0022613"],"name":["defense response","response to type I interferon","cellular response to type I interferon","type I interferon-mediated signaling pathway","RNA processing","ribonucleoprotein complex biogenesis"],"size":[583,61,57,57,725,381],"propInList":[0.20926243567753,0.60655737704918,0.614035087719298,0.614035087719298,0.0110344827586207,0.005249343832021],"log2OR":[2.40645571936024,2.82941918796788,2.86629897563625,2.86617721191944,-1.96685069350717,-2.69369495145587],"PIP":[0.999565243721008,0.582147836685181,0.206001430749893,0.205996721982956,0.563742995262146,0.348129391670227],"log2BF":[125.51383015774,39.2675521412516,37.768821025128,37.768788004459,18.3568475396669,17.6614336271113]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"52bc1ab491674748aee36c3369832e5b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-29934823c713f3ddc550","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-29934823c713f3ddc550"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions, triggered in response to the presence of a foreign body or the occurrence of an injury, which result in restriction of damage to the organism attacked or prevention/recovery from the infection caused by the attack.\"},\"text\":\"GO:0006952\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e83e1dcc507a712f7dea44fb39ea2c6b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-7be532a7fb21bcb14483","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-7be532a7fb21bcb14483"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a type I interferon stimulus. Type I interferons include the interferon-alpha, beta, delta, episilon, zeta, kappa, tau, and omega gene families.\"},\"text\":\"GO:0034340\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"790fc0a7703a67cf66ea806e099c0644"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-fcd37629850d4adba349","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-fcd37629850d4adba349"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a type I interferon stimulus. Type I interferons include the interferon-alpha, beta, delta, episilon, zeta, kappa, tau, and omega gene families.\"},\"text\":\"GO:0071357\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"02161e6b7507e936d31b1d93d54e50c7"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-28163a83a3fb84f52382","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-28163a83a3fb84f52382"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by type I interferon binding to its receptor on the surface of a target cell, and ending with the regulation of a downstream cellular process, e.g. transcription. Type I interferons include the interferon-alpha, beta, delta, episilon, zeta, kappa, tau, and omega gene families.\"},\"text\":\"GO:0060337\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"1d9ea1411941b7bceef2f3894ac6af05"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-ccb09de5516844c729c6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-ccb09de5516844c729c6"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the conversion of one or more primary RNA transcripts into one or more mature RNA molecules.\"},\"text\":\"GO:0006396\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"504da2f9c5f762c92179bc3af66f7d04"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-3cae859b2b110bfcab8c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-3cae859b2b110bfcab8c"},"children":["{\"x\":{\"opts\":{\"content\":\"A cellular process that results in the biosynthesis of constituent macromolecules, assembly, and arrangement of constituent parts of a complex containing RNA and proteins. Includes the biosynthesis of the constituent RNA and protein molecules, and those macromolecular modifications that are involved in synthesis or assembly of the ribonucleoprotein complex.\"},\"text\":\"GO:0022613\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#FF0404"},{"background":"#FF0000"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"9436adaa8cfeaa65c613553c8885833a"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd4t-cell-2" class="level3" data-number="5.3.4">
<h3 data-number="5.3.4" class="anchored" data-anchor-id="cd4t-cell-2"><span class="header-section-number">5.3.4</span> CD4T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-2d5b4c8ef3df596de624" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2d5b4c8ef3df596de624">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L2","L2","L3","L3"],"log2BF_SER":[47.4282066693421,27.8937832247683,27.8937832247683,27.8937832247683,27.8937832247683,24.9914208292728,24.9914208292728],"geneSet":["GO:0002376","GO:0043207","GO:0051707","GO:0009607","GO:0009615","GO:0019221","GO:0034097"],"name":["immune system process","response to external biotic stimulus","response to other organism","response to biotic stimulus","response to virus","cytokine-mediated signaling pathway","response to cytokine"],"size":[1292,375,375,390,186,325,524],"propInList":[0.164086687306502,0.282666666666667,0.282666666666667,0.274358974358974,0.370967741935484,0.289230769230769,0.232824427480916],"log2OR":[1.15652674476025,1.41755427302054,1.41735081754457,1.39141695168168,1.82647432628932,1.48999198970274,1.21729827629516],"PIP":[0.995056986808777,0.323031067848206,0.323022425174713,0.187492400407791,0.128513962030411,0.843796730041504,0.127999618649483],"log2BF":[61.037721466397,39.8801928677388,39.8801515919026,39.0953521043181,38.5504422733244,38.3630527286667,35.6422954471843]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-3bd5d81d89cc75f480a8","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-3bd5d81d89cc75f480a8"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"268cbbdc024d4488d878e89ff1b9523b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-a869ed022fd3505277f6","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-a869ed022fd3505277f6"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of an external biotic stimulus, an external stimulus caused by, or produced by living things.\"},\"text\":\"GO:0043207\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3d096c095c85e6af7f3a440747b5ea14"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-544cb3b0295dfb4ff53e","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-544cb3b0295dfb4ff53e"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from another living organism.\"},\"text\":\"GO:0051707\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2149a3af55e2d385df4b0510f2e7fd58"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-30052a3fdc0253bd0f7e","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-30052a3fdc0253bd0f7e"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a biotic stimulus, a stimulus caused or produced by a living organism.\"},\"text\":\"GO:0009607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6bd9d741ac345ee04e5e90714066e6a4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e7182725ae0b14522981","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e7182725ae0b14522981"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from a virus.\"},\"text\":\"GO:0009615\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-2c8ea459d8f248c5cc25","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-2c8ea459d8f248c5cc25"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-9e5d0efe88b826e2e1bc","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-9e5d0efe88b826e2e1bc"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#6BFF6B"},{"background":"#4AFF4A"},{"background":"#4AFF4A"},{"background":"#4DFF4D"},{"background":"#16FF16"},{"background":"#41FF41"},{"background":"#63FF63"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"c4dd7b2dadf714f8d01bd3c7b9520ccd"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd8t-cell-2" class="level3" data-number="5.3.5">
<h3 data-number="5.3.5" class="anchored" data-anchor-id="cd8t-cell-2"><span class="header-section-number">5.3.5</span> CD8T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-b8e9875e08fd3098ef7c" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b8e9875e08fd3098ef7c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L3","L3","L5"],"log2BF_SER":[72.7528713543139,4.93305904369295,29.5609470119374,29.5609470119374,51.5831901370366],"geneSet":["GO:0002376","GO:0006396","GO:0051607","GO:0009615","GO:0034097"],"name":["immune system process","RNA processing","defense response to virus","response to virus","response to cytokine"],"size":[1292,725,140,186,524],"propInList":[0.130030959752322,0.0096551724137931,0.385714285714286,0.327956989247312,0.204198473282443],"log2OR":[1.58702907286846,-2.07660365792973,2.25900811346952,2.03262786608676,1.80748916146058],"PIP":[0.999744474887848,0.919949054718018,0.734798550605774,0.252956330776215,0.987343490123749],"log2BF":[86.3691644360439,18.4293485459646,42.7330328250838,41.1945723407496,65.1814797114528]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"ebe78ce54ea1e0412bf7faf1e2b487af"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-c203a18327a2d7573e63","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-c203a18327a2d7573e63"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the development or functioning of the immune system, an organismal system for calibrated responses to potential internal or invasive threats.\"},\"text\":\"GO:0002376\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"1d9ea1411941b7bceef2f3894ac6af05"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-76d11d651057fcd2006e","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-76d11d651057fcd2006e"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process involved in the conversion of one or more primary RNA transcripts into one or more mature RNA molecules.\"},\"text\":\"GO:0006396\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"dece375ce5a8369e1b6f2f6ad6be3d7e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-ff7f36872ffc74a3a43a","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-ff7f36872ffc74a3a43a"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions triggered in response to the presence of a virus that act to protect the cell or organism.\"},\"text\":\"GO:0051607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6bd9d741ac345ee04e5e90714066e6a4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-3916461e409dbd64139b","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-3916461e409dbd64139b"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from a virus.\"},\"text\":\"GO:0009615\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-800bbd93f3cf6590f87c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-800bbd93f3cf6590f87c"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#34FF34"},{"background":"#FF0000"},{"background":"#00FF00"},{"background":"#00FF00"},{"background":"#18FF18"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"d22f936d59568e6c480d68bf8fa9b01b"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="mono-2" class="level3" data-number="5.3.6">
<h3 data-number="5.3.6" class="anchored" data-anchor-id="mono-2"><span class="header-section-number">5.3.6</span> Mono</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">4</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-1d35835d1e7072530c11" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1d35835d1e7072530c11">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L1","L2","L3","L3","L3"],"log2BF_SER":[26.2692501591136,26.2692501591136,26.2692501591136,14.56508083938,26.703522480246,26.703522480246,26.703522480246],"geneSet":["GO:0006952","GO:0006955","GO:0045087","GO:0022613","GO:0007154","GO:0007165","GO:0023052"],"name":["defense response","immune response","innate immune response","ribonucleoprotein complex biogenesis","cell communication","signal transduction","signaling"],"size":[583,891,359,381,2277,2123,2251],"propInList":[0.415094339622642,0.38047138047138,0.454038997214485,0.10761154855643,0.313131313131313,0.313707018370231,0.310972900932919],"log2OR":[0.958732249725352,0.785193155872235,1.16816489673547,-1.35985968309222,0.527127942781739,0.528045943176382,0.512212188442047],"PIP":[0.503160655498505,0.355438590049744,0.109085336327553,0.904163360595703,0.717582583427429,0.102317109704018,0.0851444900035858],"log2BF":[38.8950046988138,38.3935858405703,36.6894413979785,28.0363998610292,39.8414045885938,37.03130438378,36.7662419705478]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"52bc1ab491674748aee36c3369832e5b"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-6f6420202833c276bcf2","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-6f6420202833c276bcf2"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions, triggered in response to the presence of a foreign body or the occurrence of an injury, which result in restriction of damage to the organism attacked or prevention/recovery from the infection caused by the attack.\"},\"text\":\"GO:0006952\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e9165e5031ced195d9f180228fbc00f9"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-dba3602499cbf2fc1d4c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-dba3602499cbf2fc1d4c"},"children":["{\"x\":{\"opts\":{\"content\":\"Any immune system process that functions in the calibrated response of an organism to a potential internal or invasive threat.\"},\"text\":\"GO:0006955\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"bc531b0ac9400c7d40d8dffbe349325f"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-8f982b6300cdd75c04b4","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-8f982b6300cdd75c04b4"},"children":["{\"x\":{\"opts\":{\"content\":\"Innate immune responses are defense responses mediated by germline encoded components that directly recognize components of potential pathogens.\"},\"text\":\"GO:0045087\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"504da2f9c5f762c92179bc3af66f7d04"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-e3522b8fb5c29db419b4","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-e3522b8fb5c29db419b4"},"children":["{\"x\":{\"opts\":{\"content\":\"A cellular process that results in the biosynthesis of constituent macromolecules, assembly, and arrangement of constituent parts of a complex containing RNA and proteins. Includes the biosynthesis of the constituent RNA and protein molecules, and those macromolecular modifications that are involved in synthesis or assembly of the ribonucleoprotein complex.\"},\"text\":\"GO:0022613\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"7f710151e2c955127eb2a9d2877618c1"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-6dd3eb7cfa0c4dd5061d","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-6dd3eb7cfa0c4dd5061d"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that mediates interactions between a cell and its surroundings. Encompasses interactions such as signaling or attachment between one cell and another cell, between a cell and an extracellular matrix, or between a cell and any other aspect of its environment.\"},\"text\":\"GO:0007154\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3f81b6762a3e43cf473e4c3ac3a48dc8"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-a16c3d0e7dc406c2070e","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-a16c3d0e7dc406c2070e"},"children":["{\"x\":{\"opts\":{\"content\":\"The cellular process in which a signal is conveyed to trigger a change in the activity or state of a cell. Signal transduction begins with reception of a signal (e.g. a ligand binding to a receptor or receptor activation by a stimulus such as light), or for signal transduction in the absence of ligand, signal-withdrawal or the activity of a constitutively active receptor. Signal transduction ends with regulation of a downstream cellular process, e.g. regulation of transcription or regulation of a metabolic process. Signal transduction covers signaling from receptors located on the surface of the cell and signaling via molecules located within the cell. For signaling between cells, signal transduction is restricted to events at and within the receiving cell.\"},\"text\":\"GO:0007165\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"a87029249241577ddb8ad769e31ff135"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-a843863d9c7219d98ae5","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-a843863d9c7219d98ae5"},"children":["{\"x\":{\"opts\":{\"content\":\"The entirety of a process in which information is transmitted within a biological system. This process begins with an active signal and ends when a cellular response has been triggered.\"},\"text\":\"GO:0023052\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#84FF84"},{"background":"#9AFF9A"},{"background":"#6AFF6A"},{"background":"#FF5151"},{"background":"#BBFFBB"},{"background":"#BBFFBB"},{"background":"#BDFFBD"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"4b1dff9c8dfe2f820b072399f0292a9b"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="nk-2" class="level3" data-number="5.3.7">
<h3 data-number="5.3.7" class="anchored" data-anchor-id="nk-2"><span class="header-section-number">5.3.7</span> NK</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">5</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-56d9e0a85496ed26acd6" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-56d9e0a85496ed26acd6">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L2","L2","L4"],"log2BF_SER":[59.5184245002638,42.5947789522975,42.5947789522975,37.3378124446375],"geneSet":["GO:0006955","GO:0034097","GO:0019221","GO:0051607"],"name":["immune response","response to cytokine","cytokine-mediated signaling pathway","defense response to virus"],"size":[891,524,325,140],"propInList":[0.172839506172839,0.217557251908397,0.276923076923077,0.421428571428571],"log2OR":[1.56493841731309,1.63496046568128,1.88947415104013,2.42734249635958],"PIP":[0.970665156841278,0.732925772666931,0.239965200424194,0.980115950107574],"log2BF":[73.0921313066592,55.7631814195019,54.152339628795,50.9254993195776]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e9165e5031ced195d9f180228fbc00f9"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-ed1249f4c2646f6a28b5","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-ed1249f4c2646f6a28b5"},"children":["{\"x\":{\"opts\":{\"content\":\"Any immune system process that functions in the calibrated response of an organism to a potential internal or invasive threat.\"},\"text\":\"GO:0006955\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"2d700aa77c02e90e0f2b86d431a72731"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-83f10499b675726e879f","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-83f10499b675726e879f"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus.\"},\"text\":\"GO:0034097\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6aa72b3dc808cd8333385c62ba3ea6dc"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-dd7b0b120fdce7e63afb","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-dd7b0b120fdce7e63afb"},"children":["{\"x\":{\"opts\":{\"content\":\"The series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with the regulation of a downstream cellular process, e.g. transcription.\"},\"text\":\"GO:0019221\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"dece375ce5a8369e1b6f2f6ad6be3d7e"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-7cd0ecea683d61fe9a6b","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-7cd0ecea683d61fe9a6b"},"children":["{\"x\":{\"opts\":{\"content\":\"Reactions triggered in response to the presence of a virus that act to protect the cell or organism.\"},\"text\":\"GO:0051607\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#37FF37"},{"background":"#2EFF2E"},{"background":"#0EFF0E"},{"background":"#00FF00"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"b62a9960b600900c10e08e843b54dbae"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="residual-enrichment-1" class="level3" data-number="5.3.8">
<h3 data-number="5.3.8" class="anchored" data-anchor-id="residual-enrichment-1"><span class="header-section-number">5.3.8</span> Residual Enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">map_dbl</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(<span class="dv">1</span><span class="sc">/</span><span class="fu">exp</span>(susie_fits<span class="sc">$</span>fit[[.x]]<span class="sc">$</span>fit<span class="sc">$</span>lbf[<span class="dv">10</span>,]), <span class="at">method=</span><span class="st">'BH'</span>)<span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0 0 0 0 0</code></pre>
</div>
</div>
</section>
</section>
<section id="de-genes-lfsr-0.01-log2fc--0.5-down-regulated" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="de-genes-lfsr-0.01-log2fc--0.5-down-regulated"><span class="header-section-number">5.4</span> DE genes (<code>lfsr &lt; 0.01</code>, <code>log2FC &lt; -0.5</code>, down regulated)</h2>
<section id="overview-3" class="level3" data-number="5.4.1">
<h3 data-number="5.4.1" class="anchored" data-anchor-id="overview-3"><span class="header-section-number">5.4.1</span> Overview</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>make_list_and_background <span class="ot">&lt;-</span> <span class="cf">function</span>(gene, lfsr, beta){</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> gene[<span class="fu">intersect</span>(<span class="fu">which</span>(lfsr <span class="sc">&lt;</span> <span class="fl">0.01</span>), <span class="fu">which</span>(beta <span class="sc">&lt;</span> <span class="sc">-</span><span class="fl">0.5</span>))]</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">gene_list=</span>gene_list, <span class="at">gene_background=</span>gene))</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>symbol2entrez <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db, </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">unique</span>(data<span class="sc">$</span>gene_results<span class="sc">$</span>gene), </span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">'ENTREZID'</span>, </span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">'SYMBOL'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> SYMBOL)</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> data<span class="sc">$</span>long_results <span class="sc">%&gt;%</span></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(symbol2entrez) <span class="sc">%&gt;%</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">%&gt;%</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">gene_list =</span> <span class="fu">list</span>(<span class="fu">make_list_and_background</span>(ENTREZID, lfsr, beta)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the matrix</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>list_of_lists <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(gene_lists<span class="sc">$</span>gene_list, <span class="sc">~</span>.<span class="sc">$</span>gene_list)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_of_lists) <span class="ot">&lt;-</span> gene_lists<span class="sc">$</span>celltype</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>intersection_matrix <span class="ot">&lt;-</span> <span class="fu">make_intersection_matrix</span>(list_of_lists)</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(intersection_matrix, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">display_numbers =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="plague_files/figure-html/heatmap-intersection-large-effect-down-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fitting-3" class="level3" data-number="5.4.2">
<h3 data-number="5.4.2" class="anchored" data-anchor-id="fitting-3"><span class="header-section-number">5.4.2</span> Fitting</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>susie_fits <span class="ot">&lt;-</span> gene_lists <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fit =</span> <span class="fu">list</span>(gseasusie<span class="sc">::</span><span class="fu">fit_gsea_susie_webgestalt</span>(</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>gene_list, gene_list<span class="sc">$</span>gene_background,</span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">'geneontology_Biological_Process'</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  )))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>13.198 sec elapsed
7.404 sec elapsed
4.699 sec elapsed
7.36 sec elapsed
6.328 sec elapsed</code></pre>
</div>
</div>
</section>
<section id="b-cell-3" class="level3" data-number="5.4.3">
<h3 data-number="5.4.3" class="anchored" data-anchor-id="b-cell-3"><span class="header-section-number">5.4.3</span> B Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-e9ab2156d5b7c32cb04a" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e9ab2156d5b7c32cb04a">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L1","L1"],"log2BF_SER":[26.5639341891684,26.5639341891684,26.5639341891684,26.5639341891684],"geneSet":["GO:0023052","GO:0050896","GO:0007154","GO:0007165"],"name":["signaling","response to stimulus","cell communication","signal transduction"],"size":[2251,3368,2277,2123],"propInList":[0.0319857840959573,0.0273159144893112,0.0316205533596838,0.0320301460197833],"log2OR":[1.53220771109678,1.32063396607475,1.51548979355436,1.53113849495637],"PIP":[0.38061997294426,0.254034548997879,0.225895598530769,0.0759466961026192],"log2BF":[38.7870188561289,38.2036949802384,38.0343264656687,36.4617281130689]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"a87029249241577ddb8ad769e31ff135"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-d09491ac194458d130d1","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-d09491ac194458d130d1"},"children":["{\"x\":{\"opts\":{\"content\":\"The entirety of a process in which information is transmitted within a biological system. This process begins with an active signal and ends when a cellular response has been triggered.\"},\"text\":\"GO:0023052\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6b08ea515cd96f80677742cd25f2f7fb"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-8910b9482518499defe8","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-8910b9482518499defe8"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus. The process begins with detection of the stimulus and ends with a change in state or activity or the cell or organism.\"},\"text\":\"GO:0050896\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"7f710151e2c955127eb2a9d2877618c1"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-50bfc5773c69b73ed97d","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-50bfc5773c69b73ed97d"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that mediates interactions between a cell and its surroundings. Encompasses interactions such as signaling or attachment between one cell and another cell, between a cell and an extracellular matrix, or between a cell and any other aspect of its environment.\"},\"text\":\"GO:0007154\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3f81b6762a3e43cf473e4c3ac3a48dc8"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-0f446efbabdc94252a6c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-0f446efbabdc94252a6c"},"children":["{\"x\":{\"opts\":{\"content\":\"The cellular process in which a signal is conveyed to trigger a change in the activity or state of a cell. Signal transduction begins with reception of a signal (e.g. a ligand binding to a receptor or receptor activation by a stimulus such as light), or for signal transduction in the absence of ligand, signal-withdrawal or the activity of a constitutively active receptor. Signal transduction ends with regulation of a downstream cellular process, e.g. regulation of transcription or regulation of a metabolic process. Signal transduction covers signaling from receptors located on the surface of the cell and signaling via molecules located within the cell. For signaling between cells, signal transduction is restricted to events at and within the receiving cell.\"},\"text\":\"GO:0007165\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#3BFF3B"},{"background":"#56FF56"},{"background":"#3DFF3D"},{"background":"#3BFF3B"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"7a879ad40f69258cd0d2a2ac8436bd7f"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd4t-cell-3" class="level3" data-number="5.4.4">
<h3 data-number="5.4.4" class="anchored" data-anchor-id="cd4t-cell-3"><span class="header-section-number">5.4.4</span> CD4T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="cd8t-cell-3" class="level3" data-number="5.4.5">
<h3 data-number="5.4.5" class="anchored" data-anchor-id="cd8t-cell-3"><span class="header-section-number">5.4.5</span> CD8T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="mono-3" class="level3" data-number="5.4.6">
<h3 data-number="5.4.6" class="anchored" data-anchor-id="mono-3"><span class="header-section-number">5.4.6</span> Mono</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">4</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-b290317fc77f19820fde" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b290317fc77f19820fde">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L2"],"log2BF_SER":[13.5139941268683,13.5139941268683,11.3803139162214],"geneSet":["GO:0000184","GO:0006413","GO:0044281"],"name":["nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","translational initiation","small molecule metabolic process"],"size":[114,174,884],"propInList":[0.0263157894736842,0.0689655172413793,0.346153846153846],"log2OR":[-2.85004695909502,-1.98884572670866,0.660843421931668],"PIP":[0.804076790809631,0.158956959843636,0.988075852394104],"log2BF":[26.8160632532326,24.4773661185725,24.9796710314333]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"4ada8c9fe906b6ea3f31181dcbdc72ca"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-8f00dd90283a5679272f","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-8f00dd90283a5679272f"},"children":["{\"x\":{\"opts\":{\"content\":\"The nonsense-mediated decay pathway for nuclear-transcribed mRNAs degrades mRNAs in which an amino-acid codon has changed to a nonsense codon; this prevents the translation of such mRNAs into truncated, and potentially harmful, proteins.\"},\"text\":\"GO:0000184\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e646d3511df2a406123db6aa504f63d4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-f1fc0e71fcbd3df8d68a","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-f1fc0e71fcbd3df8d68a"},"children":["{\"x\":{\"opts\":{\"content\":\"The process preceding formation of the peptide bond between the first two amino acids of a protein. This includes the formation of a complex of the ribosome, mRNA or circRNA, and an initiation complex that contains the first aminoacyl-tRNA.\"},\"text\":\"GO:0006413\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"4b6dcfce6b14561654812a4ced761db4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-1a541e90c71bffd9dcdd","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-1a541e90c71bffd9dcdd"},"children":["{\"x\":{\"opts\":{\"content\":\"The chemical reactions and pathways involving small molecules, any low molecular weight, monomeric, non-encoded molecule.\"},\"text\":\"GO:0044281\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#FF0000"},{"background":"#FF0101"},{"background":"#AAFFAA"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"9ce2ce941e485f0bd0e628af5c56382e"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="nk-3" class="level3" data-number="5.4.7">
<h3 data-number="5.4.7" class="anchored" data-anchor-id="nk-3"><span class="header-section-number">5.4.7</span> NK</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">5</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="residual-enrichment-2" class="level3" data-number="5.4.8">
<h3 data-number="5.4.8" class="anchored" data-anchor-id="residual-enrichment-2"><span class="header-section-number">5.4.8</span> Residual Enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">map_dbl</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(<span class="dv">1</span><span class="sc">/</span><span class="fu">exp</span>(susie_fits<span class="sc">$</span>fit[[.x]]<span class="sc">$</span>fit<span class="sc">$</span>lbf[<span class="dv">10</span>,]), <span class="at">method=</span><span class="st">'BH'</span>)<span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0 0 0 0 0</code></pre>
</div>
</div>
</section>
</section>
<section id="unique-de-genes-lfsr-0.01" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="unique-de-genes-lfsr-0.01"><span class="header-section-number">5.5</span> Unique DE genes (<code>lfsr &lt; 0.01</code>)</h2>
<section id="overview-4" class="level3" data-number="5.5.1">
<h3 data-number="5.5.1" class="anchored" data-anchor-id="overview-4"><span class="header-section-number">5.5.1</span> Overview</h3>
<p>CD4 T-Cells ane mononucleocytes have a large number of unique DE genes, but B, CD8T, and NK cells have very few. Consequently, logistic SuSiE only reports enrichment in the prior two cell types.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to find unique elements in a list compared to others</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>find_unique_elements <span class="ot">&lt;-</span> <span class="cf">function</span>(target_list, other_lists) {</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine all other lists into a single vector</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  combined_others <span class="ot">&lt;-</span> <span class="fu">unlist</span>(other_lists)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find elements in the target list that are not in the combined vector of other lists</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setdiff</span>(target_list, combined_others)</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new list to store lists of unique elements</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>unique_elements_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">seq_along</span>(gene_lists<span class="sc">$</span>gene_list), <span class="cf">function</span>(i) {</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Current target list</span></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>  target_list <span class="ot">&lt;-</span> list_of_lists[[i]]</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All other lists except the current one</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  other_lists <span class="ot">&lt;-</span> list_of_lists[<span class="sc">-</span>i]</span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find unique elements for the current list</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">find_unique_elements</span>(target_list, other_lists)</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the matrix</span></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>list_of_lists <span class="ot">&lt;-</span> unique_elements_list</span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_of_lists) <span class="ot">&lt;-</span> gene_lists<span class="sc">$</span>celltype</span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>intersection_matrix <span class="ot">&lt;-</span> <span class="fu">make_intersection_matrix</span>(list_of_lists)</span>
<span id="cb53-23"><a href="#cb53-23" aria-hidden="true" tabindex="-1"></a>pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(intersection_matrix, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="at">display_numbers =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="plague_files/figure-html/heatmap-intersection2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fitting-4" class="level3" data-number="5.5.2">
<h3 data-number="5.5.2" class="anchored" data-anchor-id="fitting-4"><span class="header-section-number">5.5.2</span> Fitting</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign names to the resulting list of lists</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="co"># names(unique_elements_list) &lt;- names(list_of_lists)</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>unique_susie_fits <span class="ot">&lt;-</span> gene_lists <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">unique_list =</span> unique_elements_list) <span class="sc">%&gt;%</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">fit =</span> <span class="fu">list</span>(gseasusie<span class="sc">::</span><span class="fu">fit_gsea_susie_webgestalt</span>(</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>    unique_list, gene_list<span class="sc">$</span>gene_background,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">'geneontology_Biological_Process'</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  ))) <span class="sc">%&gt;%</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>4.387 sec elapsed
8.057 sec elapsed
5.302 sec elapsed
7.647 sec elapsed
4.661 sec elapsed</code></pre>
</div>
</div>
</section>
<section id="b-cell-4" class="level3" data-number="5.5.3">
<h3 data-number="5.5.3" class="anchored" data-anchor-id="b-cell-4"><span class="header-section-number">5.5.3</span> B Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-8c2cab0c27d6b2299656" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8c2cab0c27d6b2299656">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L1","L1"],"log2BF_SER":[26.5639341891684,26.5639341891684,26.5639341891684,26.5639341891684],"geneSet":["GO:0023052","GO:0050896","GO:0007154","GO:0007165"],"name":["signaling","response to stimulus","cell communication","signal transduction"],"size":[2251,3368,2277,2123],"propInList":[0.0319857840959573,0.0273159144893112,0.0316205533596838,0.0320301460197833],"log2OR":[1.53220771109678,1.32063396607475,1.51548979355436,1.53113849495637],"PIP":[0.38061997294426,0.254034548997879,0.225895598530769,0.0759466961026192],"log2BF":[38.7870188561289,38.2036949802384,38.0343264656687,36.4617281130689]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"a87029249241577ddb8ad769e31ff135"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-8a80a8ba7ee44e392529","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-8a80a8ba7ee44e392529"},"children":["{\"x\":{\"opts\":{\"content\":\"The entirety of a process in which information is transmitted within a biological system. This process begins with an active signal and ends when a cellular response has been triggered.\"},\"text\":\"GO:0023052\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"6b08ea515cd96f80677742cd25f2f7fb"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-1d63cafc2cab9cd690da","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-1d63cafc2cab9cd690da"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus. The process begins with detection of the stimulus and ends with a change in state or activity or the cell or organism.\"},\"text\":\"GO:0050896\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"7f710151e2c955127eb2a9d2877618c1"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-d9cf821d72bf3fd639ae","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-d9cf821d72bf3fd639ae"},"children":["{\"x\":{\"opts\":{\"content\":\"Any process that mediates interactions between a cell and its surroundings. Encompasses interactions such as signaling or attachment between one cell and another cell, between a cell and an extracellular matrix, or between a cell and any other aspect of its environment.\"},\"text\":\"GO:0007154\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"3f81b6762a3e43cf473e4c3ac3a48dc8"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-c437b4f0a99b898802e2","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-c437b4f0a99b898802e2"},"children":["{\"x\":{\"opts\":{\"content\":\"The cellular process in which a signal is conveyed to trigger a change in the activity or state of a cell. Signal transduction begins with reception of a signal (e.g. a ligand binding to a receptor or receptor activation by a stimulus such as light), or for signal transduction in the absence of ligand, signal-withdrawal or the activity of a constitutively active receptor. Signal transduction ends with regulation of a downstream cellular process, e.g. regulation of transcription or regulation of a metabolic process. Signal transduction covers signaling from receptors located on the surface of the cell and signaling via molecules located within the cell. For signaling between cells, signal transduction is restricted to events at and within the receiving cell.\"},\"text\":\"GO:0007165\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#3BFF3B"},{"background":"#56FF56"},{"background":"#3DFF3D"},{"background":"#3BFF3B"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"c40568f3532add2340eafb48ee370c51"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="cd4t-cell-4" class="level3" data-number="5.5.4">
<h3 data-number="5.5.4" class="anchored" data-anchor-id="cd4t-cell-4"><span class="header-section-number">5.5.4</span> CD4T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">2</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="cd8t-cell-4" class="level3" data-number="5.5.5">
<h3 data-number="5.5.5" class="anchored" data-anchor-id="cd8t-cell-4"><span class="header-section-number">5.5.5</span> CD8T Cell</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">3</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="mono-4" class="level3" data-number="5.5.6">
<h3 data-number="5.5.6" class="anchored" data-anchor-id="mono-4"><span class="header-section-number">5.5.6</span> Mono</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">4</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-1d1cd8cab058dcf60d75" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1d1cd8cab058dcf60d75">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"component":["L1","L1","L2"],"log2BF_SER":[13.5139941268683,13.5139941268683,11.3803139162214],"geneSet":["GO:0000184","GO:0006413","GO:0044281"],"name":["nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","translational initiation","small molecule metabolic process"],"size":[114,174,884],"propInList":[0.0263157894736842,0.0689655172413793,0.346153846153846],"log2OR":[-2.85004695909502,-1.98884572670866,0.660843421931668],"PIP":[0.804076790809631,0.158956959843636,0.988075852394104],"log2BF":[26.8160632532326,24.4773661185725,24.9796710314333]},"columns":[{"id":"component","name":"component","type":["glue","character"],"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'component') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['component'] === prevRow['component']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"log2BF_SER","name":"log2BF_SER","type":"numeric","aggregate":"min","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":"function(rowInfo, column, state) {\n        const firstSorted = state.sorted[0]\n        // Merge cells if unsorted or sorting by school\n        if (!firstSorted || firstSorted.id === 'log2BF_SER') {\n          const prevRow = state.pageRows[rowInfo.viewIndex - 1]\n          if (prevRow && rowInfo.values['log2BF_SER'] === prevRow['log2BF_SER']) {\n            return { visibility: 'hidden' }\n          }\n        }\n      }"},{"id":"geneSet","name":"geneSet","type":"character","cell":[{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"4ada8c9fe906b6ea3f31181dcbdc72ca"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-2978f80e8093894816ad","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-2978f80e8093894816ad"},"children":["{\"x\":{\"opts\":{\"content\":\"The nonsense-mediated decay pathway for nuclear-transcribed mRNAs degrades mRNAs in which an amino-acid codon has changed to a nonsense codon; this prevents the translation of such mRNAs into truncated, and potentially harmful, proteins.\"},\"text\":\"GO:0000184\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"e646d3511df2a406123db6aa504f63d4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-b027bdb88125105e9a75","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-b027bdb88125105e9a75"},"children":["{\"x\":{\"opts\":{\"content\":\"The process preceding formation of the peptide bond between the first two amino acids of a protein. This includes the formation of a complex of the ribosome, mRNA or circRNA, and an initiation complex that contains the first aminoacyl-tRNA.\"},\"text\":\"GO:0006413\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]},{"name":"div","attribs":{"style":{"text-decoration":"underline","text-decoration-style":"dotted","cursor":"help"}},"children":[{"name":"WidgetContainer","attribs":{"key":"4b6dcfce6b14561654812a4ced761db4"},"children":[{"name":"Fragment","attribs":[],"children":[{"name":"span","attribs":{"id":"htmlwidget-bcf0b20ef7be5014187c","width":"100%","height":"338px","className":"tippy html-widget html-fill-item"},"children":[]},{"name":"script","attribs":{"type":"application/json","data-for":"htmlwidget-bcf0b20ef7be5014187c"},"children":["{\"x\":{\"opts\":{\"content\":\"The chemical reactions and pathways involving small molecules, any low molecular weight, monomeric, non-encoded molecule.\"},\"text\":\"GO:0044281\"},\"evals\":[],\"jsHooks\":[]}"]}]}]}]}]},{"id":"name","name":"name","type":"character"},{"id":"size","name":"size","type":"numeric","format":{"cell":{"digits":0},"aggregated":{"digits":0}}},{"id":"propInList","name":"propInList","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2OR","name":"log2OR","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"style":[{"background":"#FF0000"},{"background":"#FF0101"},{"background":"#AAFFAA"}]},{"id":"PIP","name":"PIP","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"log2BF","name":"log2BF","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"groupBy":["component"],"paginateSubRows":true,"defaultExpanded":true,"dataKey":"14e7a60ba221010ba995145385b13f72"},"children":[]},"class":"reactR_markup"},"evals":["tag.attribs.columns.0.style","tag.attribs.columns.1.style"],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="nk-4" class="level3" data-number="5.5.7">
<h3 data-number="5.5.7" class="anchored" data-anchor-id="nk-4"><span class="header-section-number">5.5.7</span> NK</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>susie_fits<span class="sc">$</span>fit[[<span class="dv">5</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_cs_tbl</span>() <span class="sc">%&gt;%</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  gseasusie<span class="sc">::</span><span class="fu">make_reactable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
</div>
</section>
<section id="residual-enrichment-3" class="level3" data-number="5.5.8">
<h3 data-number="5.5.8" class="anchored" data-anchor-id="residual-enrichment-3"><span class="header-section-number">5.5.8</span> Residual Enrichment</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">map_dbl</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(<span class="dv">1</span><span class="sc">/</span><span class="fu">exp</span>(susie_fits<span class="sc">$</span>fit[[.x]]<span class="sc">$</span>fit<span class="sc">$</span>lbf[<span class="dv">10</span>,]), <span class="at">method=</span><span class="st">'BH'</span>)<span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0 0 0 0 0</code></pre>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../notebooks/pbmcs/table.html" class="pagination-link" aria-label="PBMC GSEA Table">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">PBMC GSEA Table</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../notebooks/summary.html" class="pagination-link" aria-label="Summary">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Summary</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>